--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1,5 +_,7 @@
 package org.bukkit.craftbukkit;
 
+import com.github.weepingmc.offline.CraftOfflinePlayerEditor;
+import com.github.weepingmc.offline.EditPlayer;
 import com.google.common.base.Function;
 import com.google.common.base.Joiner;
 import com.google.common.base.Preconditions;
@@ -387,6 +_,7 @@
         return ca.spottedleaf.moonrise.common.util.TickThread.isTickThread();
     }
     // Paper end - Folia reagion threading API
+    private final com.github.weepingmc.packet.PacketPipeline packetPipeline; // Weeper - packet support
 
     static {
         ConfigurationSerialization.registerClass(CraftOfflinePlayer.class);
@@ -485,6 +_,7 @@
         this.potionBrewer = new io.papermc.paper.potion.PaperPotionBrewer(console); // Paper - custom potion mixes
         datapackManager = new io.papermc.paper.datapack.PaperDatapackManager(console.getPackRepository()); // Paper
         this.spark = new io.papermc.paper.SparksFly(this); // Paper - spark
+        this.packetPipeline = new com.github.weepingmc.packet.CraftPacketPipeline(() -> ((CraftWorld)getWorlds().getFirst()).getHandle()); // Weeper - packet support
     }
 
     public boolean getCommandBlockOverride(String command) {
@@ -3066,4 +_,25 @@
     public void allowPausing(final Plugin plugin, final boolean value) {
         this.console.addPluginAllowingSleep(plugin.getName(), value);
     }
+
+    // Paper start - add disguise api
+    @Override
+    public com.destroystokyo.paper.SkinParts.@org.jetbrains.annotations.NotNull Builder newSkinPartsBuilder() {
+        return com.destroystokyo.paper.PaperSkinParts.builder();
+    }  // Paper end - add disguise api
+
+    //  Weeper start - packet support
+    @Override
+    public com.github.weepingmc.packet.PacketPipeline getPacketPipeline() {
+        return packetPipeline;
+    }
+    // Weeper end - packet support
+    // Weeper start - offline player edit
+    private final CraftOfflinePlayerEditor offlinePlayerEditor = new CraftOfflinePlayerEditor(this.getServer());
+    public CraftOfflinePlayerEditor getOfflinePlayerEditor() { return offlinePlayerEditor; }
+    @Override
+    public void editOfflinePlayer(UUID playerUUID, Consumer<EditPlayer> editPlayerConsumer) {
+        offlinePlayerEditor.editOfflinePlayer(playerUUID, editPlayerConsumer);
+    }
+    // Weeper end - offline player edit
 }
