--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -2488,4 +_,12 @@
         return POINTERS_SUPPLIER.view(this);
     }
     // Paper end
+    // Weeper start
+    @Override
+    public void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter) {
+        BlockPos position = new BlockPos(location.getBlockX(), location.getBlockY(), location.getBlockZ());
+        net.minecraft.world.level.block.entity.BlockEntity blockEntity = world.getBlockEntity(position);
+        world.blockEvent(position, blockEntity.getBlockState().getBlock(), action, actionParameter);
+    }
+    // Weeper end
 }
