--- a/src/main/java/io/papermc/paper/persistence/PersistentDataContainerView.java
+++ b/src/main/java/io/papermc/paper/persistence/PersistentDataContainerView.java
@@ -2,7 +_,7 @@
 
 import java.util.Set;
 import com.google.common.base.Preconditions;
-import org.bukkit.NamespacedKey;
+import net.kyori.adventure.key.Key;
 import org.bukkit.persistence.PersistentDataAdapterContext;
 import org.bukkit.persistence.PersistentDataContainer;
 import org.bukkit.persistence.PersistentDataHolder;
@@ -48,7 +_,7 @@
      * @throws IllegalArgumentException if the type to cast the found object to is
      * null
      */
-    <P, C> boolean has(NamespacedKey key, PersistentDataType<P, C> type);
+    <P, C> boolean has(Key key, PersistentDataType<P, C> type);
 
     /**
      * Returns if the persistent metadata provider has metadata registered matching
@@ -65,7 +_,7 @@
      * @return if a value with the provided key exists
      * @throws IllegalArgumentException if the key to look up is null
      */
-    boolean has(NamespacedKey key);
+    boolean has(Key key);
 
     /**
      * Returns if the persistent metadata provider has metadata registered
@@ -112,7 +_,7 @@
      * the {@link
      * PersistentDataType#getPrimitiveType()}
      */
-    <P, C> @Nullable C get(NamespacedKey key, PersistentDataType<P, C> type);
+    <P, C> @Nullable C get(Key key, PersistentDataType<P, C> type);
 
     /**
      * Returns the metadata value that is stored on the
@@ -135,7 +_,7 @@
      * @throws IllegalArgumentException if no suitable adapter was found for
      * the {@link PersistentDataType#getPrimitiveType()}
      */
-    <P, C> C getOrDefault(NamespacedKey key, PersistentDataType<P, C> type, C defaultValue);
+    <P, C> C getOrDefault(Key key, PersistentDataType<P, C> type, C defaultValue);
 
     /**
      * Returns the metadata value that is stored on the
@@ -186,7 +_,22 @@
      *
      * @return the key set
      */
-    Set<NamespacedKey> getKeys();
+    @Deprecated
+    default Set<org.bukkit.NamespacedKey> getKeys() {
+        return Set.of(); // NOOP: Will be rewritten via Commodore and stays as a compatibility shim
+    }
+
+    /**
+     * Get the set of keys present on this {@link PersistentDataContainer}
+     * instance.
+     * <p>
+     * Any changes made to the returned set will not be reflected on the
+     * instance.
+     *
+     * @return the key set
+     */
+    @Deprecated
+    Set<Key> keys();
 
     /**
      * Returns if the container instance is empty, therefore has no entries
