From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Tue, 16 Jun 2026 18:16:13 +0200
Subject: [PATCH] enable moonrise hook


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index e1a5d7b4bc2bdc51335b47c9d94e309e92584d8e..8e562ce5c8d2927392c4ece79c2fdaba58d2f7e0 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -5460,12 +5460,12 @@ public abstract class Entity
         return ((ServerLevel) this.level()).isPositionEntityTicking(this.blockPosition());
     }
     // Paper end
-    // Weeper start - disguise api todo update when leaf updated leaf stuff -> move leaf into feature patch
+    // Weeper start - disguise api
     public void clearPlayers() {
-        // if (trackedEntity != null) trackedEntity.moonrise$clearPlayers();
+        if (trackedEntity != null) trackedEntity.moonrise$clearPlayers();
     }
     public void updatePlayers() {
-        // if (trackedEntity != null) trackedEntity.updatePlayers(((ServerLevel)level).players());
+        if (trackedEntity != null) trackedEntity.updatePlayers(((ServerLevel)level).players());
     }
     // Weeper end - disguise api
 }
