--- a/src/main/java/io/papermc/paper/datacomponent/item/PaperPotDecorations.java
+++ b/src/main/java/io/papermc/paper/datacomponent/item/PaperPotDecorations.java
@@ -73,10 +_,10 @@
             }
 
             return new PaperPotDecorations(new net.minecraft.world.level.block.entity.PotDecorations(
-                Optional.ofNullable(this.back).map(CraftItemType::bukkitToMinecraftNew),
-                Optional.ofNullable(this.left).map(CraftItemType::bukkitToMinecraftNew),
-                Optional.ofNullable(this.right).map(CraftItemType::bukkitToMinecraftNew),
-                Optional.ofNullable(this.front).map(CraftItemType::bukkitToMinecraftNew)
+                Optional.ofNullable(this.back).map(CraftItemType::bukkitToMinecraftNewT),
+                Optional.ofNullable(this.left).map(CraftItemType::bukkitToMinecraftNewT),
+                Optional.ofNullable(this.right).map(CraftItemType::bukkitToMinecraftNewT),
+                Optional.ofNullable(this.front).map(CraftItemType::bukkitToMinecraftNewT)
             ));
         }
     }
