From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sun, 10 Mar 2024 12:22:50 +0100
Subject: [PATCH] build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index e1e434773a4c1e7cb3dd43b7bee5abe338a4eb0a..b26fb20354c46c81d8055d5fd057dcd964a6a7dd 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,8 +14,8 @@ extensions.configure(PaperweightSourceGeneratorExt::class) {
 
 dependencies {
     implementation("com.squareup:javapoet:1.13.0")
-    implementation(project(":paper-api"))
-    implementation("io.github.classgraph:classgraph:4.8.47")
+    implementation(project(":weeper-api")) // Weeper - build changes
+    implementation("io.github.classgraph:classgraph:4.8.168") // Weeper - build changes
     implementation("org.jetbrains:annotations:24.1.0")
     testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
     testRuntimeOnly("org.junit.platform:junit-platform-launcher")
