From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sun, 10 Mar 2024 12:22:50 +0100
Subject: [PATCH] build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 05d16d0b95104bd43d09203981b4248d605cc54f..ca45fd63bd4d6bd5f7e7371e2c34fedf432b0cc7 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,8 +13,8 @@ extensions.configure(PaperweightSourceGeneratorExt::class) {
 
 dependencies {
     implementation("com.squareup:javapoet:1.13.0")
-    implementation(project(":paper-api"))
-    implementation("io.github.classgraph:classgraph:4.8.47")
+    implementation(project(":weeper-api")) // Weeper - build changes
+    implementation("io.github.classgraph:classgraph:4.8.168") // Weeper - build changes
     implementation("org.jetbrains:annotations:24.0.1")
     testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
     testRuntimeOnly("org.junit.platform:junit-platform-launcher")
