From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sun, 10 Mar 2024 12:22:50 +0100
Subject: [PATCH] build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 62ceb696a547285a546f5de08636ae8060cc3ec7..925c6c0b4b36f27863e36426ce92430f06dfaa0e 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,8 +13,8 @@ extensions.configure(PaperweightSourceGeneratorExt::class) {
 
 dependencies {
     implementation("com.squareup:javapoet:1.13.0")
-    implementation(project(":paper-api"))
-    implementation("io.github.classgraph:classgraph:4.8.47")
+    implementation(project(":weeper-api")) // Weeper - build changes
+    implementation("io.github.classgraph:classgraph:4.8.168") // Weeper - build changes
     implementation("org.jetbrains:annotations:24.1.0")
     testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
     testRuntimeOnly("org.junit.platform:junit-platform-launcher")
