--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -15,11 +_,11 @@
 }
 
 dependencies {
-    minecraftJar(project(":paper-server", "mappedJarOutgoing"))
-    implementation(project(":paper-server", "macheMinecraftLibraries"))
+    minecraftJar(project(":weeper-server", "mappedJarOutgoing")) // Weeper build changes
+    implementation(project(":weeper-server", "macheMinecraftLibraries")) // Weeper build changes
 
     implementation("com.squareup:javapoet:1.13.0")
-    implementation(project(":paper-api"))
+    implementation(project(":weeper-api")) // Weeper build changes
     implementation("io.papermc.typewriter:typewriter:1.0.1") {
         isTransitive = false // paper-api already have everything
     }
