--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -7,6 +_,7 @@
 import com.google.common.collect.Lists;
 import com.mojang.logging.LogUtils;
 import io.papermc.paper.datacomponent.DataComponentType;
+import io.papermc.paper.disguise.PlayerDisguise;
 import io.papermc.paper.entity.LookAnchor;
 import io.papermc.paper.entity.TeleportFlag;
 import java.util.EnumSet;
@@ -1330,5 +_,26 @@
     public boolean hasData(final @NotNull DataComponentType type) {
         return this.entity.get(io.papermc.paper.datacomponent.PaperDataComponentType.bukkitToMinecraft(type)) != null;
     }
-
+    // Weeper start - disguise api
+    private io.papermc.paper.disguise.DisguiseData disguiseData = io.papermc.paper.disguise.DisguiseData.original();
+    @Override
+    public @org.jetbrains.annotations.NotNull io.papermc.paper.disguise.DisguiseData getDisguiseData() {
+        return disguiseData;
+    }
+    @Override
+    public void setDisguiseData(@org.jetbrains.annotations.NotNull io.papermc.paper.disguise.DisguiseData disguiseData) {
+        if (disguiseData instanceof PlayerDisguise disguise) {
+            var pose = disguise.pose();
+            if (!org.bukkit.entity.Mannequin.validPoses().contains(pose)) {
+                throw new IllegalArgumentException("Invalid pose '%s', expected one of: %s".formatted(
+                        pose.name(),
+                        org.bukkit.entity.Mannequin.validPoses() // name doesn't match
+                ));
+            }
+        }
+        getHandle().clearPlayers();
+        this.disguiseData = disguiseData;
+        getHandle().updatePlayers();
+    }
+    // Weeper end - disguise api
 }
