--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1357,4 +_,27 @@
         return this.getHandleRaw().get(PaperDataComponentType.bukkitToMinecraft(type)) != null;
     }
 
+    // Weeper start - disguise api
+    private io.papermc.paper.disguise.DisguiseData disguiseData = io.papermc.paper.disguise.DisguiseData.original();
+    @Override
+    public @org.jetbrains.annotations.NotNull io.papermc.paper.disguise.DisguiseData getDisguiseData() {
+        return disguiseData;
+    }
+    @Override
+    public void setDisguiseData(@org.jetbrains.annotations.NotNull io.papermc.paper.disguise.DisguiseData disguiseData) {
+        if (disguiseData instanceof io.papermc.paper.disguise.PlayerDisguise disguise) {
+            var pose = disguise.pose();
+            if (!org.bukkit.entity.Mannequin.validPoses().contains(pose)) {
+                throw new IllegalArgumentException("Invalid pose '%s', expected one of: %s".formatted(
+                        pose.name(),
+                        org.bukkit.entity.Mannequin.validPoses() // name doesn't match
+                ));
+            }
+        }
+        getHandle().clearPlayers();
+        this.disguiseData = disguiseData;
+        getHandle().updatePlayers();
+    }
+    // Weeper end - disguise api
+
 }
