--- a/src/main/java/org/bukkit/craftbukkit/block/CraftDecoratedPot.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftDecoratedPot.java
@@ -6,7 +_,8 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Collectors;
-import net.minecraft.world.item.Item;
+import java.util.stream.Stream;
+import net.minecraft.world.item.ItemStackTemplate;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.block.entity.DecoratedPotBlockEntity;
 import net.minecraft.world.level.block.entity.PotDecorations;
@@ -76,7 +_,7 @@
         Preconditions.checkArgument(face != null, "face must not be null");
         Preconditions.checkArgument(sherd == null || sherd == Material.BRICK || Tag.ITEMS_DECORATED_POT_SHERDS.isTagged(sherd), "sherd is not a valid sherd material: %s", sherd);
 
-        Optional<Item> sherdItem = (sherd != null) ? Optional.of(CraftItemType.bukkitToMinecraft(sherd)) : Optional.of(Items.BRICK);
+        Optional<ItemStackTemplate> sherdItem = ((sherd != null) ? Optional.of(CraftItemType.bukkitToMinecraft(sherd)) : Optional.of(Items.BRICK)).map(ItemStackTemplate::new);
         PotDecorations decorations = this.getSnapshot().getDecorations();
 
         switch (face) {
@@ -93,7 +_,7 @@
         Preconditions.checkArgument(face != null, "face must not be null");
 
         PotDecorations decorations = this.getSnapshot().getDecorations();
-        Optional<Item> sherdItem = switch (face) {
+        Optional<ItemStackTemplate> sherdItem = switch (face) {
             case BACK -> decorations.back();
             case LEFT -> decorations.left();
             case RIGHT -> decorations.right();
@@ -101,7 +_,7 @@
             default -> throw new IllegalArgumentException("Unexpected value: " + face);
         };
 
-        return CraftItemType.minecraftToBukkit(sherdItem.orElse(Items.BRICK));
+        return CraftItemType.minecraftToBukkit(sherdItem.orElse(Items.BRICK.getCraftingRemainder()));
     }
 
     @Override
@@ -109,16 +_,16 @@
         PotDecorations decorations = this.getSnapshot().getDecorations();
 
         Map<Side, Material> sherds = new EnumMap<>(Side.class);
-        sherds.put(Side.BACK, CraftItemType.minecraftToBukkit(decorations.back().orElse(Items.BRICK)));
-        sherds.put(Side.LEFT, CraftItemType.minecraftToBukkit(decorations.left().orElse(Items.BRICK)));
-        sherds.put(Side.RIGHT, CraftItemType.minecraftToBukkit(decorations.right().orElse(Items.BRICK)));
-        sherds.put(Side.FRONT, CraftItemType.minecraftToBukkit(decorations.front().orElse(Items.BRICK)));
+        sherds.put(Side.BACK, CraftItemType.minecraftToBukkit(decorations.back(), Items.BRICK));
+        sherds.put(Side.LEFT, CraftItemType.minecraftToBukkit(decorations.left(), Items.BRICK));
+        sherds.put(Side.RIGHT, CraftItemType.minecraftToBukkit(decorations.right(), Items.BRICK));
+        sherds.put(Side.FRONT, CraftItemType.minecraftToBukkit(decorations.front(), Items.BRICK));
         return sherds;
     }
 
     @Override
     public List<Material> getShards() {
-        return this.getSnapshot().getDecorations().ordered().stream().map(CraftItemType::minecraftToBukkit).collect(Collectors.toUnmodifiableList());
+        return getSherds().values().stream().toList();
     }
 
     @Override
