--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -88,6 +_,7 @@
             for (SynchedEntityData.DataItem<?> dataItem : this.itemsById) {
                 if (dataItem.isDirty()) {
                     dataItem.setDirty(false);
+                    if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, dataItem.getAccessor())) continue; // Weeper - disguise api
                     result.add(dataItem.value());
                 }
             }
@@ -115,6 +_,7 @@
     public void assignValues(final List<SynchedEntityData.DataValue<?>> items) {
         for (SynchedEntityData.DataValue<?> item : items) {
             SynchedEntityData.DataItem<?> dataItem = this.itemsById[item.id];
+            if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, dataItem.getAccessor())) continue; // Weeper - disguise api
             this.assignValue(dataItem, item);
             this.entity.onSyncedDataUpdated(dataItem.getAccessor());
         }
@@ -181,6 +_,7 @@
     public List<SynchedEntityData.DataValue<?>> packAll() {
         final List<SynchedEntityData.DataValue<?>> items = new ArrayList<>(this.itemsById.length);
         for (final DataItem<?> item : this.itemsById) {
+            if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, item.getAccessor())) continue; // Weeper - disguise api
             items.add(item.value());
         }
 
