--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -88,6 +_,7 @@
             for (SynchedEntityData.DataItem<?> dataItem : this.itemsById) {
                 if (dataItem.isDirty()) {
                     dataItem.setDirty(false);
+                    if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, dataItem.getAccessor())) continue; // Weeper - disguise api
                     list.add(dataItem.value());
                 }
             }
@@ -115,6 +_,7 @@
     public void assignValues(List<SynchedEntityData.DataValue<?>> entries) {
         for (SynchedEntityData.DataValue<?> dataValue : entries) {
             SynchedEntityData.DataItem<?> dataItem = this.itemsById[dataValue.id];
+            if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, dataItem.getAccessor())) continue; // Weeper - disguise api
             this.assignValue(dataItem, dataValue);
             this.entity.onSyncedDataUpdated(dataItem.getAccessor());
         }
@@ -181,6 +_,7 @@
     public List<SynchedEntityData.DataValue<?>> packAll() {
         final List<SynchedEntityData.DataValue<?>> list = new ArrayList<>(this.itemsById.length);
         for (final DataItem<?> dataItem : this.itemsById) {
+            if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, dataItem.getAccessor())) continue; // Weeper - disguise api
             list.add(dataItem.value());
         }
 
