--- a/net/minecraft/network/protocol/game/ClientboundSetPassengersPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetPassengersPacket.java
@@ -29,6 +_,13 @@
         this.passengers = input.readVarIntArray();
     }
 
+    // Weeper - start
+    public ClientboundSetPassengersPacket(int vehicle, int[] passengers) {
+        this.vehicle = vehicle;
+        this.passengers = passengers;
+    }
+    // Weeper - end
+
     private void write(final FriendlyByteBuf output) {
         output.writeVarInt(this.vehicle);
         output.writeVarIntArray(this.passengers);
