--- a/net/minecraft/network/protocol/game/ClientboundSetPassengersPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetPassengersPacket.java
@@ -14,6 +_,13 @@
     private final int vehicle;
     private final int[] passengers;
 
+    // Weeper - start
+    public ClientboundSetPassengersPacket(int vehicle, int[] passengers) {
+        this.vehicle = vehicle;
+        this.passengers = passengers;
+    }
+    // Weeper - end
+
     public ClientboundSetPassengersPacket(Entity vehicle) {
         this.vehicle = vehicle.getId();
         List<Entity> passengers = vehicle.getPassengers();
