--- a/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
@@ -26,6 +_,13 @@
         this.yHeadRot = input.readByte();
     }
 
+    // Weeper start - allow packet creation without real entity
+    public ClientboundRotateHeadPacket(int entityId, byte yHeadRot) {
+        this.entityId = entityId;
+        this.yHeadRot = yHeadRot;
+    }
+    // Weeper end - allow packet creation without real entity
+
     private void write(final FriendlyByteBuf output) {
         output.writeVarInt(this.entityId);
         output.writeByte(this.yHeadRot);
