From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Wed, 17 Jan 2024 22:26:49 +0100
Subject: [PATCH] add leader bonus game rule


diff --git a/src/main/java/net/minecraft/world/entity/monster/Zombie.java b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
index a12461907278cfbfa3b1c0aa74b9f07a31768b8a..8f162c9dd53437e353ce7c6b6609d611bc3a6844 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Zombie.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
@@ -601,7 +601,9 @@ public class Zombie extends Monster {
 
         if (this.random.nextFloat() < chanceMultiplier * 0.05F) {
             this.getAttribute(Attributes.SPAWN_REINFORCEMENTS_CHANCE).addOrReplacePermanentModifier(new AttributeModifier(Zombie.LEADER_ZOMBIE_BONUS_ID, this.random.nextDouble() * 0.25D + 0.5D, AttributeModifier.Operation.ADD_VALUE));
+            if(((net.minecraft.server.level.ServerLevel)this.level()).getGameRules().getRule(GameRules.RULE_LEADER_ZOMBIE_BONUS).get()){ // Weeper start - add game rule to disable leader zombie bonus
             this.getAttribute(Attributes.MAX_HEALTH).addOrReplacePermanentModifier(new AttributeModifier(Zombie.LEADER_ZOMBIE_BONUS_ID, this.random.nextDouble() * 3.0D + 1.0D, AttributeModifier.Operation.ADD_MULTIPLIED_TOTAL));
+            } // Weeper end - add game rule to disable leader zombie bonus
             this.setCanBreakDoors(true);
         }
 
diff --git a/src/main/java/net/minecraft/world/level/GameRules.java b/src/main/java/net/minecraft/world/level/GameRules.java
index 467696a78791d65a66eb380e97d33123a5a12e61..cd3bd560aa25115f2d85ba5f68c404546edc87e2 100644
--- a/src/main/java/net/minecraft/world/level/GameRules.java
+++ b/src/main/java/net/minecraft/world/level/GameRules.java
@@ -64,6 +64,7 @@ public class GameRules {
     public static final GameRules.Key<GameRules.BooleanValue> RULE_SHOWDEATHMESSAGES = GameRules.register("showDeathMessages", GameRules.Category.CHAT, GameRules.BooleanValue.create(true));
     public static final GameRules.Key<GameRules.IntegerValue> RULE_RANDOMTICKING = GameRules.register("randomTickSpeed", GameRules.Category.UPDATES, GameRules.IntegerValue.create(3));
     public static final GameRules.Key<GameRules.BooleanValue> RULE_SENDCOMMANDFEEDBACK = GameRules.register("sendCommandFeedback", GameRules.Category.CHAT, GameRules.BooleanValue.create(true));
+    public static final GameRules.Key<GameRules.BooleanValue> RULE_LEADER_ZOMBIE_BONUS = GameRules.register("leaderZombieBonus", Category.SPAWNING, GameRules.BooleanValue.create(true)); // Weeper - add game rule to disable leader zombie bonus
     public static final GameRules.Key<GameRules.BooleanValue> RULE_REDUCEDDEBUGINFO = GameRules.register("reducedDebugInfo", GameRules.Category.MISC, GameRules.BooleanValue.create(false, (minecraftserver, gamerules_gameruleboolean) -> {
         int i = gamerules_gameruleboolean.get() ? 22 : 23;
         Iterator iterator = minecraftserver.players().iterator(); // CraftBukkit - per-world
