--- a/src/main/java/org/bukkit/entity/Mannequin.java
+++ b/src/main/java/org/bukkit/entity/Mannequin.java
@@ -3,7 +_,10 @@
 import com.destroystokyo.paper.SkinParts;
 import io.papermc.paper.InternalAPIBridge;
 import io.papermc.paper.datacomponent.item.ResolvableProfile;
+import java.util.Set;
 import net.kyori.adventure.text.Component;
+import org.bukkit.Bukkit;
+import org.bukkit.UnsafeValues;
 import org.bukkit.inventory.EntityEquipment;
 import org.bukkit.inventory.MainHand;
 import org.jspecify.annotations.NullMarked;
@@ -28,6 +_,14 @@
      */
     static Component defaultDescription() {
         return InternalAPIBridge.get().defaultMannequinDescription();
+    }
+
+    /**
+     * Returns all poses valid for a {@link Mannequin}
+     * @return the set of poses
+     */
+    static Set<Pose> validPoses() {
+        return Bukkit.getUnsafe().validMannequinPoses();
     }
 
     /**
