--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -49,6 +_,7 @@
 import net.minecraft.util.ProblemReporter;
 import net.minecraft.util.datafix.DataFixers;
 import net.minecraft.util.datafix.fixes.References;
+import net.minecraft.world.entity.decoration.Mannequin;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.alchemy.Potion;
@@ -83,6 +_,7 @@
 import org.bukkit.damage.DamageType;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Pose;
 import org.bukkit.inventory.CreativeCategory;
 import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.inventory.ItemStack;
@@ -102,6 +_,9 @@
 
     private final Commodore commodore = new Commodore();
 
+    private final Set<Pose> VALID_MANNEQUIN_POSES = Mannequin.VALID_POSES.stream()
+            .map(pose -> Pose.values()[pose.ordinal()]).collect(java.util.stream.Collectors.toSet());
+
     private CraftMagicNumbers() {}
 
     @Override
@@ -617,6 +_,11 @@
         });
 
         return deserializeItem(tag);
+    }
+
+    @Override
+    public Set<Pose> validMannequinPoses() {
+        return VALID_MANNEQUIN_POSES;
     }
 
     @Override
