--- a/src/main/java/org/bukkit/GameRule.java
+++ b/src/main/java/org/bukkit/GameRule.java
@@ -466,6 +_,12 @@
     @Deprecated(forRemoval = true, since = "1.21.11")
     public static final GameRule<Integer> MINECART_MAX_SPEED = GameRules.MAX_MINECART_SPEED;
 
+    // Weeper start - custom gamerules
+    public static final GameRule<Boolean> LEADER_ZOMBIE_BONUS = getByName("leader_zombie_bonus");
+    public static final GameRule<Boolean> RULE_LEADER_ZOMBIE_BONUS = LEADER_ZOMBIE_BONUS;
+    // Weeper end - custom gamerules
+
+
     private static UnaryOperator<Boolean> inverseBool() {
         return operand -> !operand;
     }
