--- a/src/main/java/org/bukkit/GameRule.java
+++ b/src/main/java/org/bukkit/GameRule.java
@@ -330,6 +_,13 @@
      */
     public static final GameRule<Boolean> SPAWNER_BLOCKS_ENABLED = new GameRule<>("spawnerBlocksEnabled", Boolean.class);
 
+    // Weeper start - leader bonus game rule
+    /**
+     * Whether a zombie gets a leader bonus or not
+     */
+    public static final GameRule<Boolean> RULE_LEADER_ZOMBIE_BONUS = new GameRule<>("leaderZombieBonus", Boolean.class);
+    // Weeper end - leader bonus game rule
+
     // All GameRules instantiated above this for organizational purposes
     private final String name;
     private final Class<T> type;
