--- a/src/main/java/org/bukkit/GameRule.java
+++ b/src/main/java/org/bukkit/GameRule.java
@@ -466,6 +_,21 @@
     @Deprecated(forRemoval = true, since = "1.21.11")
     public static final GameRule<Integer> MINECART_MAX_SPEED = GameRules.MAX_MINECART_SPEED;
 
+    // Weeper start - custom gamerules
+    /**
+     * Represents the game rule that manages the bonus mechanics related to leader zombies.
+     * This rule determines if leader zombies grant additional functionality or mechanics
+     * when certain conditions are met.
+     *
+     * @deprecated {@link GameRules#LEADER_ZOMBIE_BONUS}
+     */
+    @Deprecated(forRemoval = true, since = "1.21.11")
+    public static final GameRule<Boolean> RULE_LEADER_ZOMBIE_BONUS = GameRules.LEADER_ZOMBIE_BONUS;
+    // Weeper end - custom gamerules
+
+
     private static UnaryOperator<Boolean> inverseBool() {
         return operand -> !operand;
     }
