--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -70,7 +_,7 @@
 /**
  * Represents a server implementation.
  */
-public interface Server extends PluginMessageRecipient, net.kyori.adventure.audience.ForwardingAudience { // Paper
+public interface Server extends PluginMessageRecipient, net.kyori.adventure.audience.ForwardingAudience, com.github.weepingmc.offline.OfflinePlayerEditor { // Paper // Weeper offline player editor
 
     /**
      * Returns the de facto plugins directory, generally used for storing plugin jars to be loaded,
@@ -2726,4 +_,20 @@
      */
     void allowPausing(@NotNull org.bukkit.plugin.Plugin plugin, boolean value);
     // Paper end - API to check if the server is sleeping
+    // Paper start - add disguise api
+    /**
+    * Creates a new skinparts builder used for overriding skin settings
+    * @return a new builder for skin parts
+    */
+    com.destroystokyo.paper.SkinParts.@NotNull Builder newSkinPartsBuilder();
+    // Paper end - add disguise api
+    // Weeper start - packet support
+    /**
+     * Returns the {@link com.github.weepingmc.packet.PacketPipeline}
+     *
+     * @return the packet pipeline
+     */
+    @NotNull
+    com.github.weepingmc.packet.PacketPipeline getPacketPipeline();
+    // Weeper end - packet support
 }
