From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Fri, 26 Nov 2021 21:05:21 +0100
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 3275a3bf46b2751813a75f9940841a78272143e5..7edbd388f140674d8030558ddbc7f878a6f1fb43 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -9,7 +9,7 @@ java {
 }
 
 dependencies {
-    implementation(project(":paper-api"))
+    implementation(project(":Weeper-API")) // Weeper
     api("com.mojang:brigadier:1.0.18")
 
     compileOnly("it.unimi.dsi:fastutil:8.5.6")
