From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Fri, 26 Nov 2021 21:05:21 +0100
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index c90209011a266e71511954f23a1e42efa8c62a54..ac1b8d2859dcd46ea85b1564b77fc536d04f888b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,7 +14,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":paper-api"))
+    implementation(project(":Weeper-API")) // Weeper
     api("com.mojang:brigadier:1.0.18")
 
     compileOnly("it.unimi.dsi:fastutil:8.5.6")
