--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemType.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemType.java
@@ -52,6 +_,19 @@
 
     private final Supplier<CraftItemMetas.ItemMetaData<M>> itemMetaData;
 
+    public static Material minecraftToBukkit(ItemStackTemplate item) {
+        return CraftMagicNumbers.getMaterial(item.typeHolder().value());
+    }
+
+    public static Material minecraftToBukkit(Optional<ItemStackTemplate> item, final Item fallback) {
+        return CraftMagicNumbers.getMaterial(item.map(ItemStackTemplate::typeHolder).map(Holder::value).orElse(fallback));
+    }
+
+    public static ItemType minecraftToBukkitNew(ItemStackTemplate minecraft) {
+        Item value = minecraft.typeHolder().value();
+        return CraftRegistry.minecraftToBukkit(value, Registries.ITEM);
+    }
+
     public static Material minecraftToBukkit(Item item) {
         return CraftMagicNumbers.getMaterial(item);
     }
@@ -62,6 +_,11 @@
 
     public static ItemType minecraftToBukkitNew(Item minecraft) {
         return CraftRegistry.minecraftToBukkit(minecraft, Registries.ITEM);
+    }
+
+
+    public static ItemStackTemplate bukkitToMinecraftNewT(ItemType bukkit) {
+        return new ItemStackTemplate(CraftRegistry.bukkitToMinecraft(bukkit));
     }
 
     public static Item bukkitToMinecraftNew(ItemType bukkit) {
