--- a/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
+++ b/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
@@ -8,9 +_,9 @@
 import java.util.Map.Entry;
 import java.util.Objects;
 import java.util.Set;
+import net.kyori.adventure.key.Key;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.Tag;
-import org.bukkit.NamespacedKey;
 import org.bukkit.craftbukkit.util.CraftNBTTagConfigSerializer;
 import org.bukkit.persistence.PersistentDataAdapterContext;
 import org.bukkit.persistence.PersistentDataContainer;
@@ -36,7 +_,7 @@
     }
 
     @Override
-    public <T, Z> void set(@NotNull NamespacedKey key, @NotNull PersistentDataType<T, Z> type, @NotNull Z value) {
+    public <T, Z> void set(@NotNull Key key, @NotNull PersistentDataType<T, Z> type, @NotNull Z value) {
         Preconditions.checkArgument(key != null, "The NamespacedKey key cannot be null");
         Preconditions.checkArgument(type != null, "The provided type cannot be null");
         Preconditions.checkArgument(value != null, "The provided value cannot be null");
@@ -44,15 +_,14 @@
         this.customDataTags.put(key.toString(), this.registry.wrap(type, type.toPrimitive(value, this.adapterContext)));
     }
 
-    @NotNull
     @Override
-    public Set<NamespacedKey> getKeys() {
-        Set<NamespacedKey> keys = new HashSet<>();
+    public Set<Key> keys() {
+        Set<Key> keys = new HashSet<>();
 
         this.customDataTags.keySet().forEach(key -> {
             String[] keyData = key.split(":", 2);
             if (keyData.length == 2) {
-                keys.add(new NamespacedKey(keyData[0], keyData[1]));
+                keys.add(Key.key(keyData[0], keyData[1]));
             }
         });
 
@@ -60,7 +_,7 @@
     }
 
     @Override
-    public void remove(@NotNull NamespacedKey key) {
+    public void remove(@NotNull Key key) {
         Preconditions.checkArgument(key != null, "The NamespacedKey key cannot be null");
 
         this.customDataTags.remove(key.toString());
