--- a/src/test/java/io/papermc/paper/world/structure/StructureSeedConfigTest.java
+++ b/src/test/java/io/papermc/paper/world/structure/StructureSeedConfigTest.java
@@ -9,6 +_,7 @@
 import net.minecraft.resources.Identifier;
 import net.minecraft.world.level.levelgen.structure.BuiltinStructureSets;
 import net.minecraft.world.level.levelgen.structure.StructureSet;
+import net.minecraft.world.level.levelgen.structure.placement.AbstractSpreadingStructurePlacement;
 import net.minecraft.world.level.levelgen.structure.placement.StructurePlacement;
 import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.support.RegistryHelper;
@@ -38,8 +_,8 @@
         for (final ResourceKey<StructureSet> setKey : structureSets.registryKeySet()) {
             assertEquals(Identifier.DEFAULT_NAMESPACE, setKey.identifier().getNamespace());
             final StructureSet set = structureSets.getValueOrThrow(setKey);
-            if (setKey == BuiltinStructureSets.STRONGHOLDS) { // special case due to seed matching world seed
-                assertEquals(0, set.placement().salt);
+            if (setKey == BuiltinStructureSets.STRONGHOLDS && set.placement() instanceof AbstractSpreadingStructurePlacement abstractSpreadingStructurePlacement) { // special case due to seed matching world seed
+                assertEquals(0, abstractSpreadingStructurePlacement.salt);
                 continue;
             }
             int salt = switch (setKey.identifier().getPath()) {
@@ -67,11 +_,16 @@
             if (setKey == BuiltinStructureSets.BURIED_TREASURES) {
                 final Field field = StructurePlacement.class.getDeclaredField("HIGHLY_ARBITRARY_RANDOM_SALT");
                 field.trySetAccessible();
-                assertEquals(0, set.placement().salt);
+                if(set.placement() instanceof AbstractSpreadingStructurePlacement abstractSpreadingStructurePlacement) {
+                    assertEquals(0, abstractSpreadingStructurePlacement.salt);
+                }
                 assertEquals(field.get(null), salt, "Mismatched default seed for " + setKey + ". Should be " + field.get(null));
                 continue;
             }
-            assertEquals(set.placement().salt, salt, "Mismatched default seed for " + setKey + ". Should be " + set.placement().salt);
+            if(set.placement() instanceof AbstractSpreadingStructurePlacement abstractSpreadingStructurePlacement) {
+                assertEquals(abstractSpreadingStructurePlacement.salt, salt, "Mismatched default seed for " + setKey + ". Should be " + abstractSpreadingStructurePlacement.salt);
+            }
+
         }
     }
 }
