--- a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+++ b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
@@ -2,6 +_,7 @@
 
 import com.google.common.base.Preconditions;
 import io.papermc.paper.persistence.PersistentDataKey;
+import net.kyori.adventure.key.Key;
 import org.bukkit.NamespacedKey;
 import org.jetbrains.annotations.NotNull;
 
@@ -32,7 +_,7 @@
      * @throws IllegalArgumentException if no suitable adapter was found for
      * the {@link PersistentDataType#getPrimitiveType()}
      */
-    <P, C> void set(@NotNull NamespacedKey key, @NotNull PersistentDataType<P, C> type, @NotNull C value);
+    <P, C> void set(@NotNull Key key, @NotNull PersistentDataType<P, C> type, @NotNull C value);
 
     /**
      * Stores a metadata value on the {@link PersistentDataHolder} instance.
@@ -48,7 +_,7 @@
      *
      * @throws IllegalArgumentException if the data key is null
      * @throws IllegalArgumentException if the value is null. Removing a tag should
-     * be done using {@link #remove(NamespacedKey)} or {@link #remove(PersistentDataKey)}
+     * be done using {@link #remove(Key)} or {@link #remove(PersistentDataKey)}
      * @throws IllegalArgumentException if no suitable adapter was found for
      * the {@link PersistentDataType#getPrimitiveType()}
      */
@@ -65,7 +_,7 @@
      *
      * @throws IllegalArgumentException if the provided key is null
      */
-    void remove(@NotNull NamespacedKey key);
+    void remove(@NotNull Key key);
 
     /**
      * Removes a custom key from the {@link PersistentDataHolder} instance.
