--- a/net/minecraft/server/network/ServerConfigurationPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerConfigurationPacketListenerImpl.java
@@ -166,7 +_,7 @@
                 // WE have to do this because the player isnt updated properly
                 serverPlayer.updateOptionsNoEvents(this.clientInformation);
             } else {
-                serverPlayer = new ServerPlayer(this.server, this.server.overworld(), this.gameProfile, this.clientInformation);
+                serverPlayer = cserver.getOfflinePlayerEditor().createOrGet(this.server, this.server.overworld(), this.gameProfile, this.clientInformation); // Weeper - Offline Player Editing
             }
             // Paper end
             playerList.placeNewPlayer(this.connection, serverPlayer, this.createCookie(this.clientInformation));
