--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -695,6 +_,7 @@
 
         final List<SynchedEntityData.DataValue<?>> values = new java.util.ArrayList<>(keys.size());
         for (final EntityDataAccessor<?> key : keys) {
+            if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip(this, key)) continue; // Weeper - disguise api
             final SynchedEntityData.DataItem<?> synchedValue = this.entityData.getItem(key);
             values.add(synchedValue.value());
         }
@@ -5435,4 +_,13 @@
         return ((ServerLevel) this.level()).isPositionEntityTicking(this.blockPosition());
     }
     // Paper end - Expose entity id counter
+    
+    // Weeper start - disguise api
+    public void clearPlayers() {
+        if (trackedEntity != null) trackedEntity.moonrise$clearPlayers();
+    }
+    public void updatePlayers() {
+        if (trackedEntity != null) trackedEntity.updatePlayers(((ServerLevel)level).players());
+    }
+    // Weeper end - disguise api
 }
