--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -613,6 +_,7 @@
 
         final List<SynchedEntityData.DataValue<?>> values = new ArrayList<>(accessors.size());
         for (final EntityDataAccessor<?> accessor : accessors) {
+            if (io.papermc.paper.disguise.DisguiseUtil.shouldSkip(this, accessor)) continue; // Weeper - disguise api
             final SynchedEntityData.DataItem<?> item = this.entityData.getItem(accessor);
             values.add(item.value());
         }
@@ -5198,4 +_,12 @@
         return ((ServerLevel) this.level()).isPositionEntityTicking(this.blockPosition());
     }
     // Paper end
+    // Weeper start - disguise api todo update when leaf updated leaf stuff -> move leaf into feature patch
+    public void clearPlayers() {
+        // if (trackedEntity != null) trackedEntity.moonrise$clearPlayers();
+    }
+    public void updatePlayers() {
+        // if (trackedEntity != null) trackedEntity.updatePlayers(((ServerLevel)level).players());
+    }
+    // Weeper end - disguise api
 }
