--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1330,6 +_,7 @@
     private void refreshDirtyAttributes() {
         Set<AttributeInstance> attributesToUpdate = this.getAttributes().getAttributesToUpdate();
 
+        if (io.papermc.paper.disguise.DisguiseUtil.shouldSkipAttributeSending(this)) return; // Weeper - disguise api
         for (AttributeInstance changedAttributeInstance : attributesToUpdate) {
             this.onAttributeUpdated(changedAttributeInstance.getAttribute());
         }
@@ -2643,6 +_,7 @@
     public boolean swing(final InteractionHand hand, final SwingAnimation animation, final boolean sendToSwingingEntity) {
         if (this.swingState.startIfAble(hand, animation, this.getModifiedSwingDuration(animation))) {
             if (this.level() instanceof ServerLevel serverLevel) {
+                if(!io.papermc.paper.disguise.DisguiseUtil.canSendAnimation(this)) return false; // Weeper - disguise api
                 ClientboundSwingAnimationPacket packet = new ClientboundSwingAnimationPacket(this, hand, animation);
                 ServerChunkCache chunkSource = serverLevel.getChunkSource();
                 if (sendToSwingingEntity) {
