--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -1,5 +_,6 @@
 package net.minecraft.world.entity;
 
+import com.destroystokyo.paper.entity.ai.PaperCustomGoal;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.Maps;
 import it.unimi.dsi.fastutil.objects.Object2IntMap.Entry;
@@ -1656,11 +_,12 @@
             () -> {
                 Set<WrappedGoal> availableGoals = this.goalSelector.getAvailableGoals();
                 List<DebugGoalInfo.DebugGoal> list = new ArrayList<>(availableGoals.size());
-                availableGoals.forEach(
-                    wrappedGoal -> list.add(
-                        new DebugGoalInfo.DebugGoal(wrappedGoal.getPriority(), wrappedGoal.isRunning(), wrappedGoal.getGoal().getClass().getSimpleName())
-                    )
-                );
+                availableGoals.forEach(wrappedGoal ->{
+                    list.add(switch (wrappedGoal.getGoal()) {
+                        case PaperCustomGoal<?> customGoal -> new DebugGoalInfo.DebugGoal(wrappedGoal.getPriority(), wrappedGoal.isRunning(), customGoal.getHandle().getClass().getSimpleName()+"*");
+                        default -> new DebugGoalInfo.DebugGoal(wrappedGoal.getPriority(), wrappedGoal.isRunning(), wrappedGoal.getGoal().getClass().getSimpleName());
+                    });
+                });
                 return new DebugGoalInfo(list);
             }
         );
