--- a/src/test/java/org/bukkit/support/provider/ClassNodeArgumentProvider.java
+++ b/src/test/java/org/bukkit/support/provider/ClassNodeArgumentProvider.java
@@ -34,7 +_,7 @@
         ClassReaderArgumentProvider classReaderArgumentProvider = new ClassReaderArgumentProvider();
         classReaderArgumentProvider.accept(new ClassReaderArguments(this.classReaderClassType(), this.excludedClasses, this.excludedPackages));
 
-        return classReaderArgumentProvider.getClassReaders().map(this::toClassNode).map(cn -> Arguments.of(cn, cn.name));
+        return classReaderArgumentProvider.getClassModels().map(cn -> Arguments.of(cn, cn.getClass().getName()));
     }
 
     private ClassReaderTest.ClassType[] classReaderClassType() {
@@ -48,14 +_,6 @@
         }
 
         return newValues;
-    }
-
-    private ClassNode toClassNode(ClassReader classReader) {
-        ClassNode classNode = new ClassNode(Opcodes.ASM9);
-
-        classReader.accept(classNode, Opcodes.ASM9);
-
-        return classNode;
     }
 
     private record ClassReaderArguments(ClassType[] value, Class<?>[] excludedClasses, String[] excludedPackages) implements ClassReaderTest {
