--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -252,6 +_,7 @@
     public static final GameRules.Key<GameRules.BooleanValue> RULE_SPAWNER_BLOCKS_ENABLED = register(
         "spawnerBlocksEnabled", GameRules.Category.MISC, GameRules.BooleanValue.create(true)
     );
+    public static final GameRules.Key<GameRules.BooleanValue> RULE_LEADER_ZOMBIE_BONUS = GameRules.register("leaderZombieBonus", Category.SPAWNING, GameRules.BooleanValue.create(true)); // Weeper - add game rule to disable leader zombie bonus
     private final Map<GameRules.Key<?>, GameRules.Value<?>> rules;
     private final FeatureFlagSet enabledFeatures;
     private final GameRules.Value<?>[] gameruleArray; // Paper - Perf: Use array for gamerule storage
