--- a/net/minecraft/world/item/ExperienceBottleItem.java
+++ b/net/minecraft/world/item/ExperienceBottleItem.java
@@ -43,6 +_,7 @@
                     0.5F,
                     0.4F / (level.getRandom().nextFloat() * 0.4F + 0.8F)
                 );
+                player.awardStat(Stats.ITEM_USED.get(this));
             } else {
                 player.containerMenu.forceHeldSlot(hand);
                 return InteractionResult.FAIL;
