--- a/src/main/java/io/papermc/paper/plugin/entrypoint/classloader/BytecodeModifyingURLClassLoader.java
+++ b/src/main/java/io/papermc/paper/plugin/entrypoint/classloader/BytecodeModifyingURLClassLoader.java
@@ -16,9 +_,6 @@
 import java.util.jar.Attributes;
 import java.util.jar.Manifest;
 import org.checkerframework.checker.nullness.qual.Nullable;
-import org.objectweb.asm.ClassReader;
-import org.objectweb.asm.ClassVisitor;
-import org.objectweb.asm.ClassWriter;
 
 import static java.util.Objects.requireNonNullElse;
 
@@ -45,16 +_,7 @@
         final URL[] urls,
         final ClassLoader parent
     ) {
-        this(urls, parent, bytes -> {
-            final ClassReader classReader = new ClassReader(bytes);
-            final ClassWriter classWriter = new ClassWriter(classReader, 0);
-            final ClassVisitor visitor = ReflectionRemapper.visitor(classWriter);
-            if (visitor == classWriter) {
-                return bytes;
-            }
-            classReader.accept(visitor, 0);
-            return classWriter.toByteArray();
-        });
+        this(urls, parent, ReflectionRemapper::processClass);
     }
 
     @Override
