--- a/paper-checkstyle/build.gradle.kts
+++ b/paper-checkstyle/build.gradle.kts
@@ -16,7 +_,7 @@
     testRuntimeOnly("org.junit.platform:junit-platform-launcher")
     testData.implementationConfigurationName("org.jspecify:jspecify:1.0.0")
 
-    checkstyle(project(":paper-checkstyle"))
+    checkstyle(project(":weeper-checkstyle")) // Weeper Checkstyle rules
 }
 
 tasks {
@@ -24,3 +_,30 @@
         useJUnitPlatform()
     }
 }
+
+tasks.named<io.papermc.paperweight.checkstyle.tasks.MergeCheckstyleConfigs>("mergeCheckstyleConfigs") {
+    overrideConfigFile.set(layout.projectDirectory.file("../paper-checkstyle/.checkstyle/checkstyle.xml"))
+}
+
+// Weeper start - Checkstyle source sets
+sourceSets {
+    main {
+        java {
+            srcDir(file("../paper-checkstyle/src/main/java"))
+        }
+        resources {
+            srcDir(file("../paper-checkstyle/src/main/resources"))
+        }
+    }
+    test {
+        java {
+            srcDir(file("../paper-checkstyle/src/test/java"))
+        }
+    }
+    named(testData.name) {
+        java {
+            srcDir(file("../paper-checkstyle/src/testData/java"))
+        }
+    }
+}
+// Weeper end - Checkstyle source sets
