--- a/src/main/java/com/destroystokyo/paper/SkinParts.java
+++ b/src/main/java/com/destroystokyo/paper/SkinParts.java
@@ -1,6 +_,7 @@
 package com.destroystokyo.paper;
 
 import io.papermc.paper.InternalAPIBridge;
+import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.Contract;
 import org.jspecify.annotations.NullMarked;
 
@@ -56,4 +_,31 @@
 
         SkinParts immutableCopy();
     }
+    // Weeper start - add disguise api
+
+    /**
+     * Builder for {@link SkinParts}.
+     * @deprecated Use {@link SkinParts.Mutable} instead.
+     */
+    @Deprecated
+    @ApiStatus.ScheduledForRemoval(inVersion = "26.2")
+    @org.jspecify.annotations.NullMarked
+    interface Builder {
+        Builder withCape(boolean cape);
+
+        Builder withJacket(boolean jacket);
+
+        Builder withLeftSleeve(boolean leftSleeve);
+
+        Builder withRightSleeve(boolean rightSleeve);
+
+        Builder withLeftPants(boolean leftPants);
+
+        Builder withRightPants(boolean rightPants);
+
+        Builder withHat(boolean hat);
+
+        SkinParts build();
+    }
+    // Weeper end - add disguise api
 }
