--- a/src/main/java/com/destroystokyo/paper/SkinParts.java
+++ b/src/main/java/com/destroystokyo/paper/SkinParts.java
@@ -56,4 +_,29 @@
 
         SkinParts immutableCopy();
     }
+    // Weeper start - add disguise api
+
+    /**
+     * Builder for {@link SkinParts}.
+     * @deprecated Use {@link SkinParts.Mutable} instead.
+     */
+    @Deprecated
+    @org.jspecify.annotations.NullMarked
+    interface Builder {
+        Builder withCape(boolean cape);
+
+        Builder withJacket(boolean jacket);
+
+        Builder withLeftSleeve(boolean leftSleeve);
+
+        Builder withRightSleeve(boolean rightSleeve);
+
+        Builder withLeftPants(boolean leftPants);
+
+        Builder withRightPants(boolean rightPants);
+
+        Builder withHat(boolean hat);
+
+        SkinParts build();
+    }
+    // Weeper end - add disguise api
 }
