--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
+++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
@@ -16,10 +_,10 @@
 import org.bukkit.scoreboard.NameTagVisibility;
 import org.bukkit.scoreboard.Team;
 
-final class CraftTeam extends CraftScoreboardComponent implements Team {
+public final class CraftTeam extends CraftScoreboardComponent implements Team { // Weeper - package-private -> public
     private final PlayerTeam team;
 
-    CraftTeam(CraftScoreboard scoreboard, PlayerTeam team) {
+    public CraftTeam(CraftScoreboard scoreboard, PlayerTeam team) { // Weeper - package-private -> public
         super(scoreboard);
         this.team = team;
     }
@@ -411,4 +_,8 @@
         final CraftTeam other = (CraftTeam) obj;
         return Objects.equals(this.team, other.team);
     }
+
+    public PlayerTeam getHandle(){ // Weeper start
+        return team;
+    } // Weeper end
 }
