--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftScoreboard.java
+++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftScoreboard.java
@@ -22,7 +_,7 @@
     final Scoreboard board;
     boolean registeredGlobally = false; // Paper - Lazily track plugin scoreboards by default
 
-    CraftScoreboard(Scoreboard board) {
+    public CraftScoreboard(Scoreboard board) { // Weeper - package private -> public
         this.board = board;
     }
 
