--- a/ca/spottedleaf/dataconverter/minecraft/versions/V4658.java
+++ b/ca/spottedleaf/dataconverter/minecraft/versions/V4658.java
@@ -147,6 +_,8 @@
                 convertBool(gameRules, "universalAnger", "minecraft:universal_anger");
                 convertBool(gameRules, "waterSourceConversion", "minecraft:water_source_conversion");
 
+                convertBool(gameRules, "leaderZombieBonus", "minecraft:leader_zombie_bonus");
+
                 return null;
             }
         });
