From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Thu, 18 Jan 2024 17:39:23 +0100
Subject: [PATCH] api generator changes


diff --git a/build.gradle.kts b/build.gradle.kts
index f90897e646ebdf28c0f972b7d760d073c7b2b72d..a1a6c08899511275666485760c576b29c7c7dd34 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -78,7 +78,7 @@ dependencies {
 }
 
 // Paper start
-val generatedApiPath: java.nio.file.Path = rootProject.projectDir.toPath().resolve("paper-api-generator/generated")
+val generatedApiPath: java.nio.file.Path = rootProject.projectDir.toPath().resolve("weeper-api-generator/generated") // Weeper - api generator changes
 idea {
     module {
         generatedSourceDirs.add(generatedApiPath.toFile())
