From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:11:50 +0200
Subject: [PATCH] startAcceptingRegistrations


diff --git a/src/main/java/org/bukkit/enchantments/Enchantment.java b/src/main/java/org/bukkit/enchantments/Enchantment.java
index 1e1c5a9d9a769018c4604e6e44fc5ed2312981e9..3843f9a98120bc63a47a68fdb5c4b06b826764b3 100644
--- a/src/main/java/org/bukkit/enchantments/Enchantment.java
+++ b/src/main/java/org/bukkit/enchantments/Enchantment.java
@@ -386,6 +386,14 @@ public abstract class Enchantment implements Keyed, net.kyori.adventure.translat
         byKey.put(enchantment.key, enchantment);
         byName.put(enchantment.getName(), enchantment);
     }
+    // Weeper start
+    /**
+     * Starts accepting any enchantment registrations
+     */
+    public static void startAcceptingRegistrations() {
+        acceptingNew = true;
+    }
+    // Weeper end
 
     /**
      * Checks if this is accepting Enchantment registrations.
