From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:11:50 +0200
Subject: [PATCH] startAcceptingRegistrations


diff --git a/src/main/java/org/bukkit/enchantments/Enchantment.java b/src/main/java/org/bukkit/enchantments/Enchantment.java
index 8183f26545998e076ad93b4e28f3a3ff693d1032..a7050b5c6660dbdc831984c3fd16e234972d6abd 100644
--- a/src/main/java/org/bukkit/enchantments/Enchantment.java
+++ b/src/main/java/org/bukkit/enchantments/Enchantment.java
@@ -415,6 +415,14 @@ public abstract class Enchantment implements Keyed, net.kyori.adventure.translat
         byKey.put(enchantment.key, enchantment);
         byName.put(enchantment.getName(), enchantment);
     }
+    // Weeper start
+    /**
+     * Starts accepting any enchantment registrations
+     */
+    public static void startAcceptingRegistrations() {
+        acceptingNew = true;
+    }
+    // Weeper end
 
     /**
      * Checks if this is accepting Enchantment registrations.
