From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Wed, 27 Dec 2023 14:51:59 +0100
Subject: [PATCH] add disguise api


diff --git a/src/main/java/com/github/weepingmc/disguise/DisguiseData.java b/src/main/java/com/github/weepingmc/disguise/DisguiseData.java
new file mode 100644
index 0000000000000000000000000000000000000000..9e6d50684ae1425e719b7acb6109875cee11725a
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/disguise/DisguiseData.java
@@ -0,0 +1,25 @@
+package com.github.weepingmc.disguise;
+
+import com.destroystokyo.paper.profile.PlayerProfile;
+import org.bukkit.entity.EntityType;
+import org.jetbrains.annotations.NotNull;
+
+public sealed interface DisguiseData permits DisguiseData.OriginalDisguise, EntityTypeDisguise, PlayerDisguise {
+    static @NotNull DisguiseData original() {
+        return new OriginalDisguise();
+    }
+
+    static @NotNull PlayerDisguise.Builder player(@NotNull PlayerProfile playerProfile) {
+        return new PlayerDisguise.Builder(playerProfile);
+    }
+
+    static @NotNull EntityTypeDisguise entity(@NotNull EntityType entityType) {
+        return new EntityTypeDisguise(entityType);
+    }
+
+    static @NotNull OriginalDisguise reset() {
+        return new OriginalDisguise();
+    }
+
+    record OriginalDisguise() implements DisguiseData{}
+}
diff --git a/src/main/java/com/github/weepingmc/disguise/EntityTypeDisguise.java b/src/main/java/com/github/weepingmc/disguise/EntityTypeDisguise.java
new file mode 100644
index 0000000000000000000000000000000000000000..d0801c94f96cdf4ecec8f540e57babb2576fb67e
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/disguise/EntityTypeDisguise.java
@@ -0,0 +1,11 @@
+package com.github.weepingmc.disguise;
+
+import java.util.Objects;
+import org.bukkit.entity.EntityType;
+import org.jetbrains.annotations.NotNull;
+
+public record EntityTypeDisguise(@NotNull EntityType entityType) implements DisguiseData {
+    public EntityTypeDisguise {
+        Objects.requireNonNull(entityType, "type cannot be null");
+    }
+}
diff --git a/src/main/java/com/github/weepingmc/disguise/PlayerDisguise.java b/src/main/java/com/github/weepingmc/disguise/PlayerDisguise.java
new file mode 100644
index 0000000000000000000000000000000000000000..a1e1a69e6286d5a1e0568e08f77d320d81b5fa3b
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/disguise/PlayerDisguise.java
@@ -0,0 +1,41 @@
+package com.github.weepingmc.disguise;
+
+import com.destroystokyo.paper.SkinParts;
+import com.destroystokyo.paper.profile.PlayerProfile;
+import java.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public record PlayerDisguise(@NotNull PlayerProfile playerProfile, boolean listed, @Nullable SkinParts skinParts) implements DisguiseData {
+
+    public  PlayerDisguise {
+        Objects.requireNonNull(playerProfile, "profile cannot be null");
+    }
+    public static @NotNull Builder builder(@NotNull PlayerProfile playerProfile) {
+        return new Builder(playerProfile);
+    }
+
+    public static class Builder {
+        private final PlayerProfile playerProfile;
+        private boolean listed;
+        private SkinParts skinParts;
+
+        public Builder(@NotNull PlayerProfile playerProfile) {
+            this.playerProfile = playerProfile;
+        }
+
+        public @NotNull Builder listed(boolean listed) {
+            this.listed = listed;
+            return this;
+        }
+
+        public @NotNull Builder skinParts(@NotNull SkinParts skinParts) {
+            this.skinParts = skinParts;
+            return this;
+        }
+
+        public @NotNull PlayerDisguise build() {
+            return new PlayerDisguise(playerProfile, listed, skinParts);
+        }
+    }
+}
diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 23def071492ccd715693d534cc506936e18f0f46..7195186db31942022c6c4c934c961e4690d51be0 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -1144,4 +1144,8 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      */
     @NotNull String getScoreboardEntryName();
     // Paper end - entity scoreboard name
+    // Weeper start - disguise api
+    @NotNull com.github.weepingmc.disguise.DisguiseData getDisguiseData();
+    void setDisuiseData(@NotNull com.github.weepingmc.disguise.DisguiseData disguiseData);
+    // Weeper end - disguise api
 }
