From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:08:41 +0200
Subject: [PATCH] trigger Block animation


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 7a439c99fc4c5ee17d674460c8e58a9fe0c64e02..f84a9c9357c2b0b00248931d44288f3f5ec92508 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -4400,4 +4401,18 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
             }
         }
     }
+    // Weeper start
+    /**
+     * Triggers a refresh of the BlockEntity.
+     * Updates the BlockEntity on the client side
+     *
+     * @param location the block location
+     * @param action the action performed
+     * @param actionParameter the parameter of that action
+     * @deprecated use the corresponding Blockentity methods
+     */
+    @Deprecated(forRemoval = true, since = "1.21.1")
+    @org.jetbrains.annotations.ApiStatus.ScheduledForRemoval(inVersion = "1.22.0")
+    void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
+    // Weeper end
 }
