From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:08:41 +0200
Subject: [PATCH] trigger Block animation


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index ef32a937e6faf1e8a5d6b1207986715bae5a246c..cc9ce8bf4e09a307a2642132a4cef25376d02762 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -4363,4 +4363,17 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
             }
         }
     }
+    // Weeper start
+    /**
+     * Triggers a refresh of the BlockEntity.
+     * Updates the BlockEntity on the client side
+     *
+     * @param location the block location
+     * @param action the action performed
+     * @param actionParameter the parameter of that action
+     * @deprecated use the corresponding Blockentity methods
+     */
+    @Deprecated
+    void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
+    // Weeper end
 }
