From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 21:36:14 +0200
Subject: [PATCH] add deprecation annotations where javadoc is bitching


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 442366ceccb9dd52a622a1c83044d1465e070e0e..cc2f711a7e6c7aafe3509a05c8be2d09fc8c2baa 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -4309,7 +4309,7 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
 
     /**
-     * @see {@link #triggerBlockAnimation(Location, int, int)}
+     * @see #triggerBlockAnimation(Location, int, int)
      */
     @Deprecated(forRemoval = true)
     default void triggerBlockEntityEvent(@NotNull Location location, int action, int actionParameter){
diff --git a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
index f0db59a556deaefefbdaca121585c0fd199c13c2..ad077dbad280331b5457b1e79232f9052aa4dfa9 100644
--- a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
+++ b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
@@ -86,6 +86,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
      *
      * @return list of available enchantment offers
      */
+    @Deprecated
     public @org.jetbrains.annotations.Nullable EnchantmentOffer @NotNull [] getOffers() { // Paper offers can contain null values
         return offers;
     }
