From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:08:41 +0200
Subject: [PATCH] trigger Block animation


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index d4c60cf8404641fa8580cb0653e6f7a10baed865..d5d6d0f78a225b9b76fe4f6387a01f301af41248 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -3982,4 +3982,23 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
             }
         }
     }
+    // Weeper start
+    /**
+     * Triggers a refresh of the BlockEntity.
+     * Updates the BlockEntity on the client side
+     *
+     * @param location the block location
+     * @param action the action performed
+     * @param actionParameter the parameter of that action
+     */
+    void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
+
+    /**
+     * @see {@link #triggerBlockAnimation(Location, int, int)}
+     */
+    @Deprecated(forRemoval = true)
+    default void triggerBlockEntityEvent(@NotNull Location location, int action, int actionParameter){
+        triggerBlockAnimation(location, action, actionParameter);
+    }
+    // Weeper end
 }
