From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:08:41 +0200
Subject: [PATCH] trigger Block animation


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 4017cc64532a9a8e42c3a6492878cd96db13fcb3..fd36aeaaef7500bb90dac7f57d418af97602f361 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -3785,4 +3785,23 @@ public interface World extends PluginMessageRecipient, Metadatable, net.kyori.ad
             }
         }
     }
+    // Weeper start
+    /**
+     * Triggers a refresh of the BlockEntity.
+     * Updates the BlockEntity on the client side
+     *
+     * @param location the block location
+     * @param action the action performed
+     * @param actionParameter the parameter of that action
+     */
+    void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
+
+    /**
+     * @see {@link #triggerBlockAnimation(Location, int, int)}
+     */
+    @Deprecated(forRemoval = true)
+    default void triggerBlockEntityEvent(@NotNull Location location, int action, int actionParameter){
+        triggerBlockAnimation(location, action, actionParameter);
+    }
+    // Weeper end
 }
