From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:08:41 +0200
Subject: [PATCH] trigger Block animation


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 35fb5047dd8a8521586a9ca9f8d70881355fb7f5..735486252fc32d139d408ff7635799abc5bf69f0 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -4376,4 +4376,17 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
             }
         }
     }
+    // Weeper start
+    /**
+     * Triggers a refresh of the BlockEntity.
+     * Updates the BlockEntity on the client side
+     *
+     * @param location the block location
+     * @param action the action performed
+     * @param actionParameter the parameter of that action
+     * @deprecated use the corresponding Blockentity methods
+     */
+    @Deprecated
+    void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
+    // Weeper end
 }
