From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 21:36:14 +0200
Subject: [PATCH] add deprecation annotations where javadoc is bitching


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index cd77b2c15dbe0443da30999360ee670f3a5a2da8..40941e78c8b32a9fe8a053a5ccc82eb9e9d1f799 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -3992,7 +3992,7 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
 
     /**
-     * @see {@link #triggerBlockAnimation(Location, int, int)}
+     * @see #triggerBlockAnimation(Location, int, int)
      */
     @Deprecated(forRemoval = true)
     default void triggerBlockEntityEvent(@NotNull Location location, int action, int actionParameter){
diff --git a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
index 8eb6f4090578d9e1b12aff813840108fdeece730..634fcb7cf4a86d532d464331e1b0c12692a76b71 100644
--- a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
+++ b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
@@ -86,6 +86,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
      *
      * @return list of available enchantment offers
      */
+    @Deprecated
     public @org.jetbrains.annotations.Nullable EnchantmentOffer @NotNull [] getOffers() { // Paper offers can contain null values
         return offers;
     }
