From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 21:36:14 +0200
Subject: [PATCH] add deprecation annotations where javadoc is bitching


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 4e269c0d0e51d654744720c3eba733adb69b2cdc..59e1880af900f4b8e38ee75b8b5f2d5a096c5c1a 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -3835,7 +3835,7 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
     void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
 
     /**
-     * @see {@link #triggerBlockAnimation(Location, int, int)}
+     * @see #triggerBlockAnimation(Location, int, int)
      */
     @Deprecated(forRemoval = true)
     default void triggerBlockEntityEvent(@NotNull Location location, int action, int actionParameter){
diff --git a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
index e669ad8ecd182c6899c7820414e6ee1f7312d699..a40c01662aada6313efddf4ecf71f30d0df55bc5 100644
--- a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
+++ b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
@@ -86,6 +86,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
      *
      * @return list of available enchantment offers
      */
+    @Deprecated
     @NotNull
     public EnchantmentOffer[] getOffers() {
         return offers;
