From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 21:36:14 +0200
Subject: [PATCH] add deprecation annotations where javadoc is bitching


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index fd36aeaaef7500bb90dac7f57d418af97602f361..e01ee0a05fc7734eb2c690b6fec03d5f1d81cc97 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -3797,7 +3797,7 @@ public interface World extends PluginMessageRecipient, Metadatable, net.kyori.ad
     void triggerBlockAnimation(@NotNull Location location, int action, int actionParameter);
 
     /**
-     * @see {@link #triggerBlockAnimation(Location, int, int)}
+     * @see #triggerBlockAnimation(Location, int, int)
      */
     @Deprecated(forRemoval = true)
     default void triggerBlockEntityEvent(@NotNull Location location, int action, int actionParameter){
diff --git a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
index e669ad8ecd182c6899c7820414e6ee1f7312d699..a40c01662aada6313efddf4ecf71f30d0df55bc5 100644
--- a/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
+++ b/src/main/java/org/bukkit/event/enchantment/PrepareItemEnchantEvent.java
@@ -86,6 +86,7 @@ public class PrepareItemEnchantEvent extends InventoryEvent implements Cancellab
      *
      * @return list of available enchantment offers
      */
+    @Deprecated
     @NotNull
     public EnchantmentOffer[] getOffers() {
         return offers;
