From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 6 Nov 2021 23:18:10 +0100
Subject: [PATCH] added executor for easy usage with CompletableFutures


diff --git a/src/main/java/org/bukkit/plugin/Plugin.java b/src/main/java/org/bukkit/plugin/Plugin.java
index 29ce61fce75606fbcc63b798fc8f13d3eafd2b74..4e2563ef430165882517f1d021242d0bc771f86d 100644
--- a/src/main/java/org/bukkit/plugin/Plugin.java
+++ b/src/main/java/org/bukkit/plugin/Plugin.java
@@ -130,6 +130,26 @@ public interface Plugin extends TabExecutor, io.papermc.paper.plugin.lifecycle.e
     @NotNull
     public Server getServer();
 
+    // Weeper start - executors
+    /**
+     * Returns an executor of the mainthread
+     * @return executor
+     */
+    @NotNull
+    default java.util.concurrent.Executor mainThreadExecutor() {
+        return (runnable) -> getServer().getScheduler().runTask(this, runnable);
+    }
+
+    /**
+     * Returns an async executor
+     * @return executor
+     */
+    @NotNull
+    default java.util.concurrent.Executor asyncExecutor() {
+        return (runnable) -> getServer().getScheduler().runTaskAsynchronously(this, runnable);
+    }
+    // Weeper stop - executors
+
     /**
      * Returns a value indicating whether or not this plugin is currently
      * enabled
