From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 6 Nov 2021 23:18:10 +0100
Subject: [PATCH] added executor for easy usage with CompletableFutures


diff --git a/src/main/java/org/bukkit/plugin/Plugin.java b/src/main/java/org/bukkit/plugin/Plugin.java
index ef431219fd2bce48bad63b6b92c99d54348d480e..66fcc77ebd1738740851df95564954781bf97d0b 100644
--- a/src/main/java/org/bukkit/plugin/Plugin.java
+++ b/src/main/java/org/bukkit/plugin/Plugin.java
@@ -117,6 +117,26 @@ public interface Plugin extends TabExecutor, io.papermc.paper.plugin.lifecycle.e
     @NotNull
     public Server getServer();
 
+    // Weeper start - executors
+    /**
+     * Returns an executor of the mainthread
+     * @return executor
+     */
+    @NotNull
+    default java.util.concurrent.Executor mainThreadExecutor() {
+        return (runnable) -> getServer().getScheduler().runTask(this, runnable);
+    }
+
+    /**
+     * Returns an async executor
+     * @return executor
+     */
+    @NotNull
+    default java.util.concurrent.Executor asyncExecutor() {
+        return (runnable) -> getServer().getScheduler().runTaskAsynchronously(this, runnable);
+    }
+    // Weeper stop - executors
+
     /**
      * Returns a value indicating whether or not this plugin is currently
      * enabled
