From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 22:30:32 +0200
Subject: [PATCH] add comments to vanilla mob goals


diff --git a/src/main/java/com/destroystokyo/paper/entity/ai/GoalKey.java b/src/main/java/com/destroystokyo/paper/entity/ai/GoalKey.java
index 9cd98c6fcfa3eb439d9013ef76ef4661175a0e5a..eb6959f760170a73ee566ace4452cbd8b9fabc26 100644
--- a/src/main/java/com/destroystokyo/paper/entity/ai/GoalKey.java
+++ b/src/main/java/com/destroystokyo/paper/entity/ai/GoalKey.java
@@ -25,11 +25,19 @@ public class GoalKey<T extends Mob> {
         this.namespacedKey = namespacedKey;
     }
 
+    /**
+     * The entity class where the goal can be applied to.
+     * @return the applicable entity
+     */
     @NotNull
     public Class<T> getEntityClass() {
         return entityClass;
     }
 
+    /**
+     * The key of that goal key
+     * @return namespaceKey
+     */
     @NotNull
     public NamespacedKey getNamespacedKey() {
         return namespacedKey;
diff --git a/src/main/java/com/destroystokyo/paper/entity/ai/VanillaGoal.java b/src/main/java/com/destroystokyo/paper/entity/ai/VanillaGoal.java
index 7921f2766f16fbd49e2afd0bba0ef30f6eefd0ce..386fa3854b991e59f46df8ce3ede74ab86f57f7b 100644
--- a/src/main/java/com/destroystokyo/paper/entity/ai/VanillaGoal.java
+++ b/src/main/java/com/destroystokyo/paper/entity/ai/VanillaGoal.java
@@ -11,189 +11,741 @@ import org.bukkit.entity.*;
  */
 public interface VanillaGoal<T extends Mob> extends Goal<T> {
 
+    /**
+     * MobGoal avoid_entity applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> AVOID_ENTITY = GoalKey.of(Creature.class, NamespacedKey.minecraft("avoid_entity"));
+    /**
+     * MobGoal beg applicable to any entity that is instance of Wolf
+     */
     GoalKey<Wolf> BEG = GoalKey.of(Wolf.class, NamespacedKey.minecraft("beg"));
+    /**
+     * MobGoal break_door applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> BREAK_DOOR = GoalKey.of(Mob.class, NamespacedKey.minecraft("break_door"));
+    /**
+     * MobGoal breath_air applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> BREATH_AIR = GoalKey.of(Creature.class, NamespacedKey.minecraft("breath_air"));
+    /**
+     * MobGoal breed applicable to any entity that is instance of Animals
+     */
     GoalKey<Animals> BREED = GoalKey.of(Animals.class, NamespacedKey.minecraft("breed"));
+    /**
+     * MobGoal cat_lie_on_bed applicable to any entity that is instance of Cat
+     */
     GoalKey<Cat> CAT_LIE_ON_BED = GoalKey.of(Cat.class, NamespacedKey.minecraft("cat_lie_on_bed"));
+    /**
+     * MobGoal cat_sit_on_block applicable to any entity that is instance of Cat
+     */
     GoalKey<Cat> CAT_SIT_ON_BLOCK = GoalKey.of(Cat.class, NamespacedKey.minecraft("cat_sit_on_block"));
+    /**
+     * MobGoal dolphin_jump applicable to any entity that is instance of Dolphin
+     */
     GoalKey<Dolphin> DOLPHIN_JUMP = GoalKey.of(Dolphin.class, NamespacedKey.minecraft("dolphin_jump"));
+    /**
+     * MobGoal eat_block applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> EAT_BLOCK = GoalKey.of(Mob.class, NamespacedKey.minecraft("eat_block"));
+    /**
+     * MobGoal flee_sun applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> FLEE_SUN = GoalKey.of(Creature.class, NamespacedKey.minecraft("flee_sun"));
+    /**
+     * MobGoal float applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> FLOAT = GoalKey.of(Mob.class, NamespacedKey.minecraft("float"));
+    /**
+     * MobGoal follow_boat applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> FOLLOW_BOAT = GoalKey.of(Creature.class, NamespacedKey.minecraft("follow_boat"));
+    /**
+     * MobGoal follow_flock_leader applicable to any entity that is instance of Fish
+     */
     GoalKey<Fish> FOLLOW_FLOCK_LEADER = GoalKey.of(Fish.class, NamespacedKey.minecraft("follow_flock_leader"));
+    /**
+     * MobGoal follow_mob applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> FOLLOW_MOB = GoalKey.of(Mob.class, NamespacedKey.minecraft("follow_mob"));
+    /**
+     * MobGoal follow_owner applicable to any entity that is instance of Tameable
+     */
     GoalKey<Tameable> FOLLOW_OWNER = GoalKey.of(Tameable.class, NamespacedKey.minecraft("follow_owner"));
+    /**
+     * MobGoal follow_parent applicable to any entity that is instance of Animals
+     */
     GoalKey<Animals> FOLLOW_PARENT = GoalKey.of(Animals.class, NamespacedKey.minecraft("follow_parent"));
+    /**
+     * MobGoal golem_random_stroll_in_village applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> GOLEM_RANDOM_STROLL_IN_VILLAGE = GoalKey.of(Creature.class, NamespacedKey.minecraft("golem_random_stroll_in_village"));
+    /**
+     * MobGoal interact applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> INTERACT = GoalKey.of(Mob.class, NamespacedKey.minecraft("interact"));
+    /**
+     * MobGoal land_on_owners_shoulder applicable to any entity that is instance of Parrot
+     */
     GoalKey<Parrot> LAND_ON_OWNERS_SHOULDER = GoalKey.of(Parrot.class, NamespacedKey.minecraft("land_on_owners_shoulder"));
+    /**
+     * MobGoal leap_at applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> LEAP_AT = GoalKey.of(Mob.class, NamespacedKey.minecraft("leap_at"));
+    /**
+     * MobGoal llama_follow_caravan applicable to any entity that is instance of Llama
+     */
     GoalKey<Llama> LLAMA_FOLLOW_CARAVAN = GoalKey.of(Llama.class, NamespacedKey.minecraft("llama_follow_caravan"));
+    /**
+     * MobGoal look_at_player applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> LOOK_AT_PLAYER = GoalKey.of(Mob.class, NamespacedKey.minecraft("look_at_player"));
+    /**
+     * MobGoal look_at_trading_player applicable to any entity that is instance of AbstractVillager
+     */
     GoalKey<AbstractVillager> LOOK_AT_TRADING_PLAYER = GoalKey.of(AbstractVillager.class, NamespacedKey.minecraft("look_at_trading_player"));
+    /**
+     * MobGoal melee_attack applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> MELEE_ATTACK = GoalKey.of(Creature.class, NamespacedKey.minecraft("melee_attack"));
+    /**
+     * MobGoal move_back_to_village applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> MOVE_BACK_TO_VILLAGE = GoalKey.of(Creature.class, NamespacedKey.minecraft("move_back_to_village"));
+    /**
+     * MobGoal move_through_village applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> MOVE_THROUGH_VILLAGE = GoalKey.of(Creature.class, NamespacedKey.minecraft("move_through_village"));
+    /**
+     * MobGoal move_towards_restriction applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> MOVE_TOWARDS_RESTRICTION = GoalKey.of(Creature.class, NamespacedKey.minecraft("move_towards_restriction"));
+    /**
+     * MobGoal move_towards applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> MOVE_TOWARDS = GoalKey.of(Creature.class, NamespacedKey.minecraft("move_towards"));
+    /**
+     * MobGoal ocelot_attack applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> OCELOT_ATTACK = GoalKey.of(Mob.class, NamespacedKey.minecraft("ocelot_attack"));
+    /**
+     * MobGoal offer_flower applicable to any entity that is instance of IronGolem
+     */
     GoalKey<IronGolem> OFFER_FLOWER = GoalKey.of(IronGolem.class, NamespacedKey.minecraft("offer_flower"));
+    /**
+     * MobGoal open_door applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> OPEN_DOOR = GoalKey.of(Mob.class, NamespacedKey.minecraft("open_door"));
+    /**
+     * MobGoal panic applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> PANIC = GoalKey.of(Creature.class, NamespacedKey.minecraft("panic"));
+    /**
+     * MobGoal pathfind_to_raid applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> PATHFIND_TO_RAID = GoalKey.of(Raider.class, NamespacedKey.minecraft("pathfind_to_raid"));
+    /**
+     * MobGoal random_look_around applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> RANDOM_LOOK_AROUND = GoalKey.of(Mob.class, NamespacedKey.minecraft("random_look_around"));
+    /**
+     * MobGoal random_stroll applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> RANDOM_STROLL = GoalKey.of(Creature.class, NamespacedKey.minecraft("random_stroll"));
+    /**
+     * MobGoal random_swimming applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> RANDOM_SWIMMING = GoalKey.of(Creature.class, NamespacedKey.minecraft("random_swimming"));
+    /**
+     * MobGoal ranged_attack applicable to any entity that is instance of RangedEntity
+     */
     GoalKey<RangedEntity> RANGED_ATTACK = GoalKey.of(RangedEntity.class, NamespacedKey.minecraft("ranged_attack"));
+    /**
+     * MobGoal ranged_bow_attack applicable to any entity that is instance of Monster
+     */
     GoalKey<Monster> RANGED_BOW_ATTACK = GoalKey.of(Monster.class, NamespacedKey.minecraft("ranged_bow_attack"));
+    /**
+     * MobGoal ranged_crossbow_attack applicable to any entity that is instance of Monster
+     */
     GoalKey<Monster> RANGED_CROSSBOW_ATTACK = GoalKey.of(Monster.class, NamespacedKey.minecraft("ranged_crossbow_attack"));
+    /**
+     * MobGoal remove_block applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> REMOVE_BLOCK = GoalKey.of(Creature.class, NamespacedKey.minecraft("remove_block"));
+    /**
+     * MobGoal restrict_sun applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> RESTRICT_SUN = GoalKey.of(Creature.class, NamespacedKey.minecraft("restrict_sun"));
+    /**
+     * MobGoal run_around_like_crazy applicable to any entity that is instance of AbstractHorse
+     */
     GoalKey<AbstractHorse> RUN_AROUND_LIKE_CRAZY = GoalKey.of(AbstractHorse.class, NamespacedKey.minecraft("run_around_like_crazy"));
+    /**
+     * MobGoal sit_when_ordered_to applicable to any entity that is instance of Tameable
+     */
     GoalKey<Tameable> SIT_WHEN_ORDERED_TO = GoalKey.of(Tameable.class, NamespacedKey.minecraft("sit_when_ordered_to"));
+    /**
+     * MobGoal stroll_through_village applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> STROLL_THROUGH_VILLAGE = GoalKey.of(Creature.class, NamespacedKey.minecraft("stroll_through_village"));
+    /**
+     * MobGoal swell applicable to any entity that is instance of Creeper
+     */
     GoalKey<Creeper> SWELL = GoalKey.of(Creeper.class, NamespacedKey.minecraft("swell"));
+    /**
+     * MobGoal tempt applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> TEMPT = GoalKey.of(Creature.class, NamespacedKey.minecraft("tempt"));
+    /**
+     * MobGoal trade_with_player applicable to any entity that is instance of AbstractVillager
+     */
     GoalKey<AbstractVillager> TRADE_WITH_PLAYER = GoalKey.of(AbstractVillager.class, NamespacedKey.minecraft("trade_with_player"));
+    /**
+     * MobGoal try_find_water applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> TRY_FIND_WATER = GoalKey.of(Creature.class, NamespacedKey.minecraft("try_find_water"));
+    /**
+     * MobGoal use_item applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> USE_ITEM = GoalKey.of(Mob.class, NamespacedKey.minecraft("use_item"));
+    /**
+     * MobGoal water_avoiding_random_flying applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> WATER_AVOIDING_RANDOM_FLYING = GoalKey.of(Creature.class, NamespacedKey.minecraft("water_avoiding_random_flying"));
+    /**
+     * MobGoal water_avoiding_random_stroll applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> WATER_AVOIDING_RANDOM_STROLL = GoalKey.of(Creature.class, NamespacedKey.minecraft("water_avoiding_random_stroll"));
+    /**
+     * MobGoal zombie_attack applicable to any entity that is instance of Zombie
+     */
     GoalKey<Zombie> ZOMBIE_ATTACK = GoalKey.of(Zombie.class, NamespacedKey.minecraft("zombie_attack"));
+    /**
+     * MobGoal defend_village applicable to any entity that is instance of IronGolem
+     */
     GoalKey<IronGolem> DEFEND_VILLAGE = GoalKey.of(IronGolem.class, NamespacedKey.minecraft("defend_village"));
+    /**
+     * MobGoal hurt_by applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> HURT_BY = GoalKey.of(Creature.class, NamespacedKey.minecraft("hurt_by"));
+    /**
+     * MobGoal nearest_attackable applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> NEAREST_ATTACKABLE = GoalKey.of(Mob.class, NamespacedKey.minecraft("nearest_attackable"));
+    /**
+     * MobGoal nearest_attackable_witch applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> NEAREST_ATTACKABLE_WITCH = GoalKey.of(Raider.class, NamespacedKey.minecraft("nearest_attackable_witch"));
+    /**
+     * MobGoal nearest_healable_raider applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> NEAREST_HEALABLE_RAIDER = GoalKey.of(Raider.class, NamespacedKey.minecraft("nearest_healable_raider"));
+    /**
+     * MobGoal non_tame_random applicable to any entity that is instance of Tameable
+     */
     GoalKey<Tameable> NON_TAME_RANDOM = GoalKey.of(Tameable.class, NamespacedKey.minecraft("non_tame_random"));
+    /**
+     * MobGoal owner_hurt_by applicable to any entity that is instance of Tameable
+     */
     GoalKey<Tameable> OWNER_HURT_BY = GoalKey.of(Tameable.class, NamespacedKey.minecraft("owner_hurt_by"));
+    /**
+     * MobGoal owner_hurt applicable to any entity that is instance of Tameable
+     */
     GoalKey<Tameable> OWNER_HURT = GoalKey.of(Tameable.class, NamespacedKey.minecraft("owner_hurt"));
+    /**
+     * MobGoal reset_universal_anger applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> RESET_UNIVERSAL_ANGER = GoalKey.of(Mob.class, NamespacedKey.minecraft("reset_universal_anger"));
+    /**
+     * MobGoal fish_swim applicable to any entity that is instance of Fish
+     */
     GoalKey<Fish> FISH_SWIM = GoalKey.of(Fish.class, NamespacedKey.minecraft("fish_swim"));
+    /**
+     * MobGoal bee_attack applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_ATTACK = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_attack"));
+    /**
+     * MobGoal bee_become_angry applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_BECOME_ANGRY = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_become_angry"));
+    /**
+     * MobGoal bee_enter_hive applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_ENTER_HIVE = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_enter_hive"));
+    /**
+     * MobGoal bee_go_to_hive applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_GO_TO_HIVE = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_go_to_hive"));
+    /**
+     * MobGoal bee_go_to_known_flower applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_GO_TO_KNOWN_FLOWER = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_go_to_known_flower"));
+    /**
+     * MobGoal bee_grow_crop applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_GROW_CROP = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_grow_crop"));
+    /**
+     * MobGoal bee_hurt_by_other applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_HURT_BY_OTHER = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_hurt_by_other"));
+    /**
+     * MobGoal bee_locate_hive applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_LOCATE_HIVE = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_locate_hive"));
+    /**
+     * MobGoal bee_pollinate applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_POLLINATE = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_pollinate"));
+    /**
+     * MobGoal bee_wander applicable to any entity that is instance of Bee
+     */
     GoalKey<Bee> BEE_WANDER = GoalKey.of(Bee.class, NamespacedKey.minecraft("bee_wander"));
+    /**
+     * MobGoal cat_avoid_entity applicable to any entity that is instance of Cat
+     */
     GoalKey<Cat> CAT_AVOID_ENTITY = GoalKey.of(Cat.class, NamespacedKey.minecraft("cat_avoid_entity"));
+    /**
+     * MobGoal cat_relax_on_owner applicable to any entity that is instance of Cat
+     */
     GoalKey<Cat> CAT_RELAX_ON_OWNER = GoalKey.of(Cat.class, NamespacedKey.minecraft("cat_relax_on_owner"));
+    /**
+     * MobGoal cat_tempt applicable to any entity that is instance of Cat
+     */
     GoalKey<Cat> CAT_TEMPT = GoalKey.of(Cat.class, NamespacedKey.minecraft("cat_tempt"));
+    /**
+     * MobGoal dolphin_swim_to_treasure applicable to any entity that is instance of Dolphin
+     */
     GoalKey<Dolphin> DOLPHIN_SWIM_TO_TREASURE = GoalKey.of(Dolphin.class, NamespacedKey.minecraft("dolphin_swim_to_treasure"));
+    /**
+     * MobGoal dolphin_swim_with_player applicable to any entity that is instance of Dolphin
+     */
     GoalKey<Dolphin> DOLPHIN_SWIM_WITH_PLAYER = GoalKey.of(Dolphin.class, NamespacedKey.minecraft("dolphin_swim_with_player"));
+    /**
+     * MobGoal play_with_items applicable to any entity that is instance of Dolphin
+     */
     GoalKey<Dolphin> PLAY_WITH_ITEMS = GoalKey.of(Dolphin.class, NamespacedKey.minecraft("play_with_items"));
+    /**
+     * MobGoal defend_trusted applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> DEFEND_TRUSTED = GoalKey.of(Fox.class, NamespacedKey.minecraft("defend_trusted"));
+    /**
+     * MobGoal faceplant applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FACEPLANT = GoalKey.of(Fox.class, NamespacedKey.minecraft("faceplant"));
+    /**
+     * MobGoal fox_breed applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_BREED = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_breed"));
+    /**
+     * MobGoal fox_eat_berries applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_EAT_BERRIES = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_eat_berries"));
+    /**
+     * MobGoal fox_float applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_FLOAT = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_float"));
+    /**
+     * MobGoal fox_follow_parent applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_FOLLOW_PARENT = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_follow_parent"));
+    /**
+     * MobGoal fox_look_at_player applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_LOOK_AT_PLAYER = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_look_at_player"));
+    /**
+     * MobGoal fox_melee_attack applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_MELEE_ATTACK = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_melee_attack"));
+    /**
+     * MobGoal fox_panic applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_PANIC = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_panic"));
+    /**
+     * MobGoal fox_pounce applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_POUNCE = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_pounce"));
+    /**
+     * MobGoal fox_search_for_items applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_SEARCH_FOR_ITEMS = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_search_for_items"));
+    /**
+     * MobGoal fox_stroll_through_village applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> FOX_STROLL_THROUGH_VILLAGE = GoalKey.of(Fox.class, NamespacedKey.minecraft("fox_stroll_through_village"));
+    /**
+     * MobGoal perch_and_search applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> PERCH_AND_SEARCH = GoalKey.of(Fox.class, NamespacedKey.minecraft("perch_and_search"));
+    /**
+     * MobGoal seek_shelter applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> SEEK_SHELTER = GoalKey.of(Fox.class, NamespacedKey.minecraft("seek_shelter"));
+    /**
+     * MobGoal sleep applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> SLEEP = GoalKey.of(Fox.class, NamespacedKey.minecraft("sleep"));
+    /**
+     * MobGoal stalk_prey applicable to any entity that is instance of Fox
+     */
     GoalKey<Fox> STALK_PREY = GoalKey.of(Fox.class, NamespacedKey.minecraft("stalk_prey"));
+    /**
+     * MobGoal ocelot_avoid_entity applicable to any entity that is instance of Ocelot
+     */
     GoalKey<Ocelot> OCELOT_AVOID_ENTITY = GoalKey.of(Ocelot.class, NamespacedKey.minecraft("ocelot_avoid_entity"));
+    /**
+     * MobGoal ocelot_tempt applicable to any entity that is instance of Ocelot
+     */
     GoalKey<Ocelot> OCELOT_TEMPT = GoalKey.of(Ocelot.class, NamespacedKey.minecraft("ocelot_tempt"));
+    /**
+     * MobGoal panda_attack applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_ATTACK = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_attack"));
+    /**
+     * MobGoal panda_avoid applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_AVOID = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_avoid"));
+    /**
+     * MobGoal panda_breed applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_BREED = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_breed"));
+    /**
+     * MobGoal panda_hurt_by applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_HURT_BY = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_hurt_by"));
+    /**
+     * MobGoal panda_lie_on_back applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_LIE_ON_BACK = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_lie_on_back"));
+    /**
+     * MobGoal panda_look_at_player applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_LOOK_AT_PLAYER = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_look_at_player"));
+    /**
+     * MobGoal panda_panic applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_PANIC = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_panic"));
+    /**
+     * MobGoal panda_roll applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_ROLL = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_roll"));
+    /**
+     * MobGoal panda_sit applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_SIT = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_sit"));
+    /**
+     * MobGoal panda_sneeze applicable to any entity that is instance of Panda
+     */
     GoalKey<Panda> PANDA_SNEEZE = GoalKey.of(Panda.class, NamespacedKey.minecraft("panda_sneeze"));
+    /**
+     * MobGoal polar_bear_attack_players applicable to any entity that is instance of PolarBear
+     */
     GoalKey<PolarBear> POLAR_BEAR_ATTACK_PLAYERS = GoalKey.of(PolarBear.class, NamespacedKey.minecraft("polar_bear_attack_players"));
+    /**
+     * MobGoal polar_bear_hurt_by applicable to any entity that is instance of PolarBear
+     */
     GoalKey<PolarBear> POLAR_BEAR_HURT_BY = GoalKey.of(PolarBear.class, NamespacedKey.minecraft("polar_bear_hurt_by"));
+    /**
+     * MobGoal polar_bear_melee_attack applicable to any entity that is instance of PolarBear
+     */
     GoalKey<PolarBear> POLAR_BEAR_MELEE_ATTACK = GoalKey.of(PolarBear.class, NamespacedKey.minecraft("polar_bear_melee_attack"));
+    /**
+     * MobGoal polar_bear_panic applicable to any entity that is instance of PolarBear
+     */
     GoalKey<PolarBear> POLAR_BEAR_PANIC = GoalKey.of(PolarBear.class, NamespacedKey.minecraft("polar_bear_panic"));
+    /**
+     * MobGoal pufferfish_puff applicable to any entity that is instance of PufferFish
+     */
     GoalKey<PufferFish> PUFFERFISH_PUFF = GoalKey.of(PufferFish.class, NamespacedKey.minecraft("pufferfish_puff"));
+    /**
+     * MobGoal evil_rabbit_attack applicable to any entity that is instance of Rabbit
+     */
     GoalKey<Rabbit> EVIL_RABBIT_ATTACK = GoalKey.of(Rabbit.class, NamespacedKey.minecraft("evil_rabbit_attack"));
+    /**
+     * MobGoal rabbit_avoid_entity applicable to any entity that is instance of Rabbit
+     */
     GoalKey<Rabbit> RABBIT_AVOID_ENTITY = GoalKey.of(Rabbit.class, NamespacedKey.minecraft("rabbit_avoid_entity"));
+    /**
+     * MobGoal rabbit_panic applicable to any entity that is instance of Rabbit
+     */
     GoalKey<Rabbit> RABBIT_PANIC = GoalKey.of(Rabbit.class, NamespacedKey.minecraft("rabbit_panic"));
+    /**
+     * MobGoal raid_garden applicable to any entity that is instance of Rabbit
+     */
     GoalKey<Rabbit> RAID_GARDEN = GoalKey.of(Rabbit.class, NamespacedKey.minecraft("raid_garden"));
+    /**
+     * MobGoal squid_flee applicable to any entity that is instance of Squid
+     */
     GoalKey<Squid> SQUID_FLEE = GoalKey.of(Squid.class, NamespacedKey.minecraft("squid_flee"));
+    /**
+     * MobGoal squid_random_movement applicable to any entity that is instance of Squid
+     */
     GoalKey<Squid> SQUID_RANDOM_MOVEMENT = GoalKey.of(Squid.class, NamespacedKey.minecraft("squid_random_movement"));
+    /**
+     * MobGoal turtle_breed applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_BREED = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_breed"));
+    /**
+     * MobGoal turtle_go_home applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_GO_HOME = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_go_home"));
+    /**
+     * MobGoal turtle_go_to_water applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_GO_TO_WATER = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_go_to_water"));
+    /**
+     * MobGoal turtle_lay_egg applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_LAY_EGG = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_lay_egg"));
+    /**
+     * MobGoal turtle_panic applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_PANIC = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_panic"));
+    /**
+     * MobGoal turtle_random_stroll applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_RANDOM_STROLL = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_random_stroll"));
+    /**
+     * MobGoal turtle_travel applicable to any entity that is instance of Turtle
+     */
     GoalKey<Turtle> TURTLE_TRAVEL = GoalKey.of(Turtle.class, NamespacedKey.minecraft("turtle_travel"));
+    /**
+     * MobGoal wolf_avoid_entity applicable to any entity that is instance of Wolf
+     */
     GoalKey<Wolf> WOLF_AVOID_ENTITY = GoalKey.of(Wolf.class, NamespacedKey.minecraft("wolf_avoid_entity"));
+    /**
+     * MobGoal llama_attack_wolf applicable to any entity that is instance of Llama
+     */
     GoalKey<Llama> LLAMA_ATTACK_WOLF = GoalKey.of(Llama.class, NamespacedKey.minecraft("llama_attack_wolf"));
+    /**
+     * MobGoal llama_hurt_by applicable to any entity that is instance of Llama
+     */
     GoalKey<Llama> LLAMA_HURT_BY = GoalKey.of(Llama.class, NamespacedKey.minecraft("llama_hurt_by"));
+    /**
+     * MobGoal skeleton_trap applicable to any entity that is instance of SkeletonHorse
+     */
     GoalKey<SkeletonHorse> SKELETON_TRAP = GoalKey.of(SkeletonHorse.class, NamespacedKey.minecraft("skeleton_trap"));
+    /**
+     * MobGoal trader_llama_defend_wandering_trader applicable to any entity that is instance of Llama
+     */
     GoalKey<Llama> TRADER_LLAMA_DEFEND_WANDERING_TRADER = GoalKey.of(Llama.class, NamespacedKey.minecraft("trader_llama_defend_wandering_trader"));
+    /**
+     * MobGoal wither_do_nothing applicable to any entity that is instance of Wither
+     */
     GoalKey<Wither> WITHER_DO_NOTHING = GoalKey.of(Wither.class, NamespacedKey.minecraft("wither_do_nothing"));
+    /**
+     * MobGoal raider_open_door applicable to any entity that is instance of Illager
+     */
     GoalKey<Illager> RAIDER_OPEN_DOOR = GoalKey.of(Illager.class, NamespacedKey.minecraft("raider_open_door"));
+    /**
+     * MobGoal skeleton_melee applicable to any entity that is instance of Skeleton
+     */
     GoalKey<Skeleton> SKELETON_MELEE = GoalKey.of(Skeleton.class, NamespacedKey.minecraft("skeleton_melee"));
+    /**
+     * MobGoal blaze_attack applicable to any entity that is instance of Blaze
+     */
     GoalKey<Blaze> BLAZE_ATTACK = GoalKey.of(Blaze.class, NamespacedKey.minecraft("blaze_attack"));
+    /**
+     * MobGoal drowned_attack applicable to any entity that is instance of Drowned
+     */
     GoalKey<Drowned> DROWNED_ATTACK = GoalKey.of(Drowned.class, NamespacedKey.minecraft("drowned_attack"));
+    /**
+     * MobGoal drowned_go_to_beach applicable to any entity that is instance of Drowned
+     */
     GoalKey<Drowned> DROWNED_GO_TO_BEACH = GoalKey.of(Drowned.class, NamespacedKey.minecraft("drowned_go_to_beach"));
+    /**
+     * MobGoal drowned_go_to_water applicable to any entity that is instance of Creature
+     */
     GoalKey<Creature> DROWNED_GO_TO_WATER = GoalKey.of(Creature.class, NamespacedKey.minecraft("drowned_go_to_water"));
+    /**
+     * MobGoal drowned_swim_up applicable to any entity that is instance of Drowned
+     */
     GoalKey<Drowned> DROWNED_SWIM_UP = GoalKey.of(Drowned.class, NamespacedKey.minecraft("drowned_swim_up"));
+    /**
+     * MobGoal drowned_trident_attack applicable to any entity that is instance of RangedEntity
+     */
     GoalKey<RangedEntity> DROWNED_TRIDENT_ATTACK = GoalKey.of(RangedEntity.class, NamespacedKey.minecraft("drowned_trident_attack"));
+    /**
+     * MobGoal enderman_freeze_when_looked_at applicable to any entity that is instance of Enderman
+     */
     GoalKey<Enderman> ENDERMAN_FREEZE_WHEN_LOOKED_AT = GoalKey.of(Enderman.class, NamespacedKey.minecraft("enderman_freeze_when_looked_at"));
+    /**
+     * MobGoal enderman_leave_block applicable to any entity that is instance of Enderman
+     */
     GoalKey<Enderman> ENDERMAN_LEAVE_BLOCK = GoalKey.of(Enderman.class, NamespacedKey.minecraft("enderman_leave_block"));
+    /**
+     * MobGoal enderman_look_for_player applicable to any entity that is instance of Enderman
+     */
     GoalKey<Enderman> ENDERMAN_LOOK_FOR_PLAYER = GoalKey.of(Enderman.class, NamespacedKey.minecraft("enderman_look_for_player"));
+    /**
+     * MobGoal enderman_take_block applicable to any entity that is instance of Enderman
+     */
     GoalKey<Enderman> ENDERMAN_TAKE_BLOCK = GoalKey.of(Enderman.class, NamespacedKey.minecraft("enderman_take_block"));
+    /**
+     * MobGoal evoker_attack_spell applicable to any entity that is instance of Evoker
+     */
     GoalKey<Evoker> EVOKER_ATTACK_SPELL = GoalKey.of(Evoker.class, NamespacedKey.minecraft("evoker_attack_spell"));
+    /**
+     * MobGoal evoker_casting_spell applicable to any entity that is instance of Evoker
+     */
     GoalKey<Evoker> EVOKER_CASTING_SPELL = GoalKey.of(Evoker.class, NamespacedKey.minecraft("evoker_casting_spell"));
+    /**
+     * MobGoal evoker_summon_spell applicable to any entity that is instance of Evoker
+     */
     GoalKey<Evoker> EVOKER_SUMMON_SPELL = GoalKey.of(Evoker.class, NamespacedKey.minecraft("evoker_summon_spell"));
+    /**
+     * MobGoal evoker_wololo_spell applicable to any entity that is instance of Evoker
+     */
     GoalKey<Evoker> EVOKER_WOLOLO_SPELL = GoalKey.of(Evoker.class, NamespacedKey.minecraft("evoker_wololo_spell"));
+    /**
+     * MobGoal ghast_look applicable to any entity that is instance of Ghast
+     */
     GoalKey<Ghast> GHAST_LOOK = GoalKey.of(Ghast.class, NamespacedKey.minecraft("ghast_look"));
+    /**
+     * MobGoal ghast_shoot_fireball applicable to any entity that is instance of Ghast
+     */
     GoalKey<Ghast> GHAST_SHOOT_FIREBALL = GoalKey.of(Ghast.class, NamespacedKey.minecraft("ghast_shoot_fireball"));
+    /**
+     * MobGoal random_float_around applicable to any entity that is instance of Ghast
+     */
     GoalKey<Ghast> RANDOM_FLOAT_AROUND = GoalKey.of(Ghast.class, NamespacedKey.minecraft("random_float_around"));
+    /**
+     * MobGoal guardian_attack applicable to any entity that is instance of Guardian
+     */
     GoalKey<Guardian> GUARDIAN_ATTACK = GoalKey.of(Guardian.class, NamespacedKey.minecraft("guardian_attack"));
+    /**
+     * MobGoal illusioner_blindness_spell applicable to any entity that is instance of Illusioner
+     */
     GoalKey<Illusioner> ILLUSIONER_BLINDNESS_SPELL = GoalKey.of(Illusioner.class, NamespacedKey.minecraft("illusioner_blindness_spell"));
+    /**
+     * MobGoal illusioner_mirror_spell applicable to any entity that is instance of Illusioner
+     */
     GoalKey<Illusioner> ILLUSIONER_MIRROR_SPELL = GoalKey.of(Illusioner.class, NamespacedKey.minecraft("illusioner_mirror_spell"));
+    /**
+     * MobGoal long_distance_patrol applicable to any entity that is instance of Monster
+     */
     GoalKey<Monster> LONG_DISTANCE_PATROL = GoalKey.of(Monster.class, NamespacedKey.minecraft("long_distance_patrol"));
+    /**
+     * MobGoal phantom_attack_player applicable to any entity that is instance of Phantom
+     */
     GoalKey<Phantom> PHANTOM_ATTACK_PLAYER = GoalKey.of(Phantom.class, NamespacedKey.minecraft("phantom_attack_player"));
+    /**
+     * MobGoal phantom_attack_strategy applicable to any entity that is instance of Phantom
+     */
     GoalKey<Phantom> PHANTOM_ATTACK_STRATEGY = GoalKey.of(Phantom.class, NamespacedKey.minecraft("phantom_attack_strategy"));
+    /**
+     * MobGoal phantom_circle_around_anchor applicable to any entity that is instance of Phantom
+     */
     GoalKey<Phantom> PHANTOM_CIRCLE_AROUND_ANCHOR = GoalKey.of(Phantom.class, NamespacedKey.minecraft("phantom_circle_around_anchor"));
+    /**
+     * MobGoal phantom_sweep_attack applicable to any entity that is instance of Phantom
+     */
     GoalKey<Phantom> PHANTOM_SWEEP_ATTACK = GoalKey.of(Phantom.class, NamespacedKey.minecraft("phantom_sweep_attack"));
+    /**
+     * MobGoal ravager_melee_attack applicable to any entity that is instance of Ravager
+     */
     GoalKey<Ravager> RAVAGER_MELEE_ATTACK = GoalKey.of(Ravager.class, NamespacedKey.minecraft("ravager_melee_attack"));
+    /**
+     * MobGoal shulker_attack applicable to any entity that is instance of Shulker
+     */
     GoalKey<Shulker> SHULKER_ATTACK = GoalKey.of(Shulker.class, NamespacedKey.minecraft("shulker_attack"));
+    /**
+     * MobGoal shulker_defense_attack applicable to any entity that is instance of Shulker
+     */
     GoalKey<Shulker> SHULKER_DEFENSE_ATTACK = GoalKey.of(Shulker.class, NamespacedKey.minecraft("shulker_defense_attack"));
+    /**
+     * MobGoal shulker_nearest_attack applicable to any entity that is instance of Shulker
+     */
     GoalKey<Shulker> SHULKER_NEAREST_ATTACK = GoalKey.of(Shulker.class, NamespacedKey.minecraft("shulker_nearest_attack"));
+    /**
+     * MobGoal shulker_peek applicable to any entity that is instance of Shulker
+     */
     GoalKey<Shulker> SHULKER_PEEK = GoalKey.of(Shulker.class, NamespacedKey.minecraft("shulker_peek"));
+    /**
+     * MobGoal silverfish_merge_with_stone applicable to any entity that is instance of Silverfish
+     */
     GoalKey<Silverfish> SILVERFISH_MERGE_WITH_STONE = GoalKey.of(Silverfish.class, NamespacedKey.minecraft("silverfish_merge_with_stone"));
+    /**
+     * MobGoal silverfish_wake_up_friends applicable to any entity that is instance of Silverfish
+     */
     GoalKey<Silverfish> SILVERFISH_WAKE_UP_FRIENDS = GoalKey.of(Silverfish.class, NamespacedKey.minecraft("silverfish_wake_up_friends"));
+    /**
+     * MobGoal slime_attack applicable to any entity that is instance of Slime
+     */
     GoalKey<Slime> SLIME_ATTACK = GoalKey.of(Slime.class, NamespacedKey.minecraft("slime_attack"));
+    /**
+     * MobGoal slime_float applicable to any entity that is instance of Slime
+     */
     GoalKey<Slime> SLIME_FLOAT = GoalKey.of(Slime.class, NamespacedKey.minecraft("slime_float"));
+    /**
+     * MobGoal slime_keep_on_jumping applicable to any entity that is instance of Slime
+     */
     GoalKey<Slime> SLIME_KEEP_ON_JUMPING = GoalKey.of(Slime.class, NamespacedKey.minecraft("slime_keep_on_jumping"));
+    /**
+     * MobGoal slime_random_direction applicable to any entity that is instance of Slime
+     */
     GoalKey<Slime> SLIME_RANDOM_DIRECTION = GoalKey.of(Slime.class, NamespacedKey.minecraft("slime_random_direction"));
+    /**
+     * MobGoal spellcaster_casting_spell applicable to any entity that is instance of Spellcaster
+     */
     GoalKey<Spellcaster> SPELLCASTER_CASTING_SPELL = GoalKey.of(Spellcaster.class, NamespacedKey.minecraft("spellcaster_casting_spell"));
+    /**
+     * MobGoal spider_attack applicable to any entity that is instance of Spider
+     */
     GoalKey<Spider> SPIDER_ATTACK = GoalKey.of(Spider.class, NamespacedKey.minecraft("spider_attack"));
+    /**
+     * MobGoal spider applicable to any entity that is instance of Spider
+     */
     GoalKey<Spider> SPIDER = GoalKey.of(Spider.class, NamespacedKey.minecraft("spider"));
+    /**
+     * MobGoal strider_go_to_lava applicable to any entity that is instance of Strider
+     */
     GoalKey<Strider> STRIDER_GO_TO_LAVA = GoalKey.of(Strider.class, NamespacedKey.minecraft("strider_go_to_lava"));
+    /**
+     * MobGoal vex_charge_attack applicable to any entity that is instance of Vex
+     */
     GoalKey<Vex> VEX_CHARGE_ATTACK = GoalKey.of(Vex.class, NamespacedKey.minecraft("vex_charge_attack"));
+    /**
+     * MobGoal vex_copy_owner applicable to any entity that is instance of Vex
+     */
     GoalKey<Vex> VEX_COPY_OWNER = GoalKey.of(Vex.class, NamespacedKey.minecraft("vex_copy_owner"));
+    /**
+     * MobGoal vex_random_move applicable to any entity that is instance of Vex
+     */
     GoalKey<Vex> VEX_RANDOM_MOVE = GoalKey.of(Vex.class, NamespacedKey.minecraft("vex_random_move"));
+    /**
+     * MobGoal vindicator_break_door applicable to any entity that is instance of Mob
+     */
     GoalKey<Mob> VINDICATOR_BREAK_DOOR = GoalKey.of(Mob.class, NamespacedKey.minecraft("vindicator_break_door"));
+    /**
+     * MobGoal vindicator_johnny_attack applicable to any entity that is instance of Vindicator
+     */
     GoalKey<Vindicator> VINDICATOR_JOHNNY_ATTACK = GoalKey.of(Vindicator.class, NamespacedKey.minecraft("vindicator_johnny_attack"));
+    /**
+     * MobGoal vindicator_melee_attack applicable to any entity that is instance of Vindicator
+     */
     GoalKey<Vindicator> VINDICATOR_MELEE_ATTACK = GoalKey.of(Vindicator.class, NamespacedKey.minecraft("vindicator_melee_attack"));
+    /**
+     * MobGoal zombie_attack_turtle_egg applicable to any entity that is instance of Zombie
+     */
     GoalKey<Zombie> ZOMBIE_ATTACK_TURTLE_EGG = GoalKey.of(Zombie.class, NamespacedKey.minecraft("zombie_attack_turtle_egg"));
+    /**
+     * MobGoal wander_to_position applicable to any entity that is instance of WanderingTrader
+     */
     GoalKey<WanderingTrader> WANDER_TO_POSITION = GoalKey.of(WanderingTrader.class, NamespacedKey.minecraft("wander_to_position"));
+    /**
+     * MobGoal hold_ground_attack applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> HOLD_GROUND_ATTACK = GoalKey.of(Raider.class, NamespacedKey.minecraft("hold_ground_attack"));
+    /**
+     * MobGoal obtain_raid_leader_banner applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> OBTAIN_RAID_LEADER_BANNER = GoalKey.of(Raider.class, NamespacedKey.minecraft("obtain_raid_leader_banner"));
+    /**
+     * MobGoal raider_celebration applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> RAIDER_CELEBRATION = GoalKey.of(Raider.class, NamespacedKey.minecraft("raider_celebration"));
+    /**
+     * MobGoal raider_move_through_village applicable to any entity that is instance of Raider
+     */
     GoalKey<Raider> RAIDER_MOVE_THROUGH_VILLAGE = GoalKey.of(Raider.class, NamespacedKey.minecraft("raider_move_through_village"));
 
     // Purpur start
