From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Fri, 10 Sep 2021 22:58:04 +0200
Subject: [PATCH] add checkerframework dependency


diff --git a/build.gradle.kts b/build.gradle.kts
index 02f80c9a34a23c2285a1e05b41f53b4bafc28c5a..2190d41569196fb5fdf75bdbdd35bb47d19ca741 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -42,6 +42,7 @@ dependencies {
 
     implementation("org.ow2.asm:asm:9.1")
     implementation("org.ow2.asm:asm-commons:9.1")
+    implementation("org.checkerframework:checker-qual:3.6.1")
 
     compileOnly("org.apache.maven:maven-resolver-provider:3.8.1")
     compileOnly("org.apache.maven.resolver:maven-resolver-connector-basic:1.7.0")
