From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 6 Nov 2021 23:18:10 +0100
Subject: [PATCH] added executor for easy usage with CompletableFutures


diff --git a/src/main/java/org/bukkit/plugin/Plugin.java b/src/main/java/org/bukkit/plugin/Plugin.java
index 4eb639fbb46a0848be207149ea433455550fae1c..3a72e97a6c5ce06c5be2d16f93d8fcb44fe2c127 100644
--- a/src/main/java/org/bukkit/plugin/Plugin.java
+++ b/src/main/java/org/bukkit/plugin/Plugin.java
@@ -2,12 +2,14 @@ package org.bukkit.plugin;
 
 import java.io.File;
 import java.io.InputStream;
+import java.util.concurrent.Executor;
 import java.util.logging.Logger;
 import org.bukkit.Server;
 import org.bukkit.command.TabExecutor;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.generator.BiomeProvider;
 import org.bukkit.generator.ChunkGenerator;
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -117,6 +119,22 @@ public interface Plugin extends TabExecutor {
     @NotNull
     public Server getServer();
 
+    // Weeper start - executors
+    /**
+     * Returns an executor of the mainthread
+     * @return executor
+     */
+    @NotNull
+    Executor mainThreadExecutor();
+
+    /**
+     * Returns an async executor
+     * @return executor
+     */
+    @NonNull
+    Executor asyncExecutor();
+    // Weeper stop - executors
+
     /**
      * Returns a value indicating whether or not this plugin is currently
      * enabled
diff --git a/src/main/java/org/bukkit/plugin/java/JavaPlugin.java b/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
index a536c47fb5f05cad046c2f3374c69f4f83f7f32c..e9bab4a6175c900379e1e0bce2a0b0a07258e5b9 100644
--- a/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
+++ b/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
@@ -12,6 +12,7 @@ import java.io.Reader;
 import java.net.URL;
 import java.net.URLConnection;
 import java.util.List;
+import java.util.concurrent.Executor;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import org.bukkit.Server;
@@ -26,6 +27,7 @@ import org.bukkit.plugin.PluginBase;
 import org.bukkit.plugin.PluginDescriptionFile;
 import org.bukkit.plugin.PluginLoader;
 import org.bukkit.plugin.PluginLogger;
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -102,6 +104,18 @@ public abstract class JavaPlugin extends PluginBase {
         return server;
     }
 
+    // Weeper start - executors
+    @Override
+    public @NotNull Executor mainThreadExecutor() {
+        return (runnable) -> getServer().getScheduler().runTask(this, runnable);
+    }
+
+    @Override
+    public @NonNull Executor asyncExecutor() {
+        return (runnable) -> getServer().getScheduler().runTaskAsynchronously(this, runnable);
+    }
+    // Weeper stop - executors
+
     /**
      * Returns a value indicating whether or not this plugin is currently
      * enabled
diff --git a/src/test/java/org/bukkit/plugin/TestPlugin.java b/src/test/java/org/bukkit/plugin/TestPlugin.java
index 43b58e920e739bb949ac0673e9ef73ba7b500dc9..d56bfe14a83b3df7fb64e4f16cc09cf42513a8e8 100644
--- a/src/test/java/org/bukkit/plugin/TestPlugin.java
+++ b/src/test/java/org/bukkit/plugin/TestPlugin.java
@@ -1,14 +1,18 @@
 package org.bukkit.plugin;
 
-import java.io.File;
-import java.io.InputStream;
-import java.util.List;
 import org.bukkit.Server;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
 import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.generator.BiomeProvider;
 import org.bukkit.generator.ChunkGenerator;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.Executor;
 
 public class TestPlugin extends PluginBase {
     private boolean enabled = true;
@@ -84,6 +88,16 @@ public class TestPlugin extends PluginBase {
         throw new UnsupportedOperationException("Not supported.");
     }
 
+    @Override
+    public @NotNull Executor mainThreadExecutor() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    @Override
+    public @NonNull Executor asyncExecutor() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
     @Override
     public boolean isEnabled() {
         return enabled;
