From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Wed, 10 Nov 2021 18:46:52 +0100
Subject: [PATCH] Added an overload with Criteria constants


diff --git a/src/main/java/org/bukkit/scoreboard/Scoreboard.java b/src/main/java/org/bukkit/scoreboard/Scoreboard.java
index 880b20bf25f74e9cb54ff3fb282a5b74db6a0a85..ade75793c03c7a89af1f73ac02986b9ed5ae4866 100644
--- a/src/main/java/org/bukkit/scoreboard/Scoreboard.java
+++ b/src/main/java/org/bukkit/scoreboard/Scoreboard.java
@@ -49,6 +49,29 @@ public interface Scoreboard {
     @NotNull
     @Deprecated
     Objective registerNewObjective(@NotNull String name, @NotNull String criteria, @Nullable net.kyori.adventure.text.Component displayName) throws IllegalArgumentException;
+
+    /**
+     * Registers an Objective on this Scoreboard
+     *
+     * @param name Name of the Objective
+     * @param criteria {@link Criteria} for the Objective
+     * @param displayName Name displayed to players for the Objective.
+     * @return The registered Objective
+     * @throws IllegalArgumentException if name is null
+     * @throws IllegalArgumentException if name is longer than 16
+     *     characters.
+     * @throws IllegalArgumentException if criteria is null
+     * @throws IllegalArgumentException if displayName is null
+     * @throws IllegalArgumentException if displayName is longer than 128
+     *     characters.
+     * @throws IllegalArgumentException if an objective by that name already
+     *     exists
+     */
+    @NotNull
+    default Objective registerNewObjective(@NotNull String name, @NotNull Criteria criteria, @Nullable net.kyori.adventure.text.Component displayName) throws IllegalArgumentException {
+        return registerNewObjective(name, criteria.value(), displayName);
+    }
+
     /**
      * Registers an Objective on this Scoreboard
      *
diff --git a/src/main/src/main/java/org/bukkit/scoreboard/Criteria.java b/src/main/src/main/java/org/bukkit/scoreboard/Criteria.java
new file mode 100644
index 0000000000000000000000000000000000000000..fb5b629bbfe59421e1fbe30f7c80049bd77e2d5e
--- /dev/null
+++ b/src/main/src/main/java/org/bukkit/scoreboard/Criteria.java
@@ -0,0 +1,17 @@
+package org.bukkit.scoreboard;
+
+import org.jetbrains.annotations.NotNull;
+
+public record Criteria(@NotNull String value) {
+    public static final Criteria DUMMY = new Criteria("dummy");
+    public static final Criteria TRIGGER = new Criteria("trigger");
+    public static final Criteria DEATH = new Criteria("deathCount");
+    public static final Criteria PLAYER_KILLS = new Criteria("playerKillCount");
+    public static final Criteria TOTAL_KILLS = new Criteria("totalKillCount");
+    public static final Criteria HEALTH = new Criteria("health");
+    public static final Criteria XP = new Criteria("xp");
+    public static final Criteria LEVEL = new Criteria("level");
+    public static final Criteria FOOD = new Criteria("food");
+    public static final Criteria AIR = new Criteria("air");
+    public static final Criteria ARMOR = new Criteria("armor");
+}
