From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Mon, 16 Oct 2023 18:10:11 +0200
Subject: [PATCH] add player abilities using sealed classes


diff --git a/src/main/java/com/github/weepingmc/packet/PacketStepBuilder.java b/src/main/java/com/github/weepingmc/packet/PacketStepBuilder.java
index e4db4fa403a1fec235d33b6e16c9bb1d21e09f46..0877b45695ddc2795869f41d9cf937d049b85609 100644
--- a/src/main/java/com/github/weepingmc/packet/PacketStepBuilder.java
+++ b/src/main/java/com/github/weepingmc/packet/PacketStepBuilder.java
@@ -101,6 +101,15 @@ public interface PacketStepBuilder {
     @NotNull
     PacketStepBuilder playerAbilities(@NotNull Set<PlayerAbility> playerAbilities, float flySpeed, float fieldOfViewModifier);
 
+    /**
+     * sets the player abilities
+     *
+     * @param playerAbilities the abilities to modify
+     * @return builder
+     */
+    @NotNull
+    PacketStepBuilder withPlayerAbilities(@NotNull Set<com.github.weepingmc.packet.options.abilities.PlayerAbility> playerAbilities);
+
     @NotNull
     PacketStepBuilder showTestMarker(@NotNull Location location, @NotNull Color color, @Nullable String text, int time);
 
diff --git a/src/main/java/com/github/weepingmc/packet/options/abilities/FlySpeedAbility.java b/src/main/java/com/github/weepingmc/packet/options/abilities/FlySpeedAbility.java
new file mode 100644
index 0000000000000000000000000000000000000000..8d28e3e1ddd662fb15e0e9033c770d2775263ad0
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/packet/options/abilities/FlySpeedAbility.java
@@ -0,0 +1,4 @@
+package com.github.weepingmc.packet.options.abilities;
+
+public record FlySpeedAbility(float speed) implements PlayerAbility{
+}
diff --git a/src/main/java/com/github/weepingmc/packet/options/abilities/PlayerAbility.java b/src/main/java/com/github/weepingmc/packet/options/abilities/PlayerAbility.java
new file mode 100644
index 0000000000000000000000000000000000000000..557b73cbdf1c2e1a31db83368ecf39a1fd53704b
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/packet/options/abilities/PlayerAbility.java
@@ -0,0 +1,32 @@
+package com.github.weepingmc.packet.options.abilities;
+
+public sealed interface PlayerAbility permits AllowFlying, CreativeModeInstantBreak, WalkSpeedAbility, FlySpeedAbility, Flying, Invulnerable {
+    PlayerAbility ALLOW_FLYING = new AllowFlying();
+    PlayerAbility CREATIVE_MODE_INSTANT_BREAK = new CreativeModeInstantBreak();
+    PlayerAbility FLYING = new Flying();
+    PlayerAbility INVULNERABLE = new Invulnerable();
+
+    static PlayerAbility flySpeedAbilityOf(float speed){
+        return new FlySpeedAbility(speed);
+    }
+
+    static PlayerAbility walkSpeedAbilityOf(float speed){
+        return new WalkSpeedAbility(speed);
+    }
+}
+
+final class  AllowFlying implements PlayerAbility {
+    AllowFlying(){}
+}
+
+final class CreativeModeInstantBreak implements PlayerAbility {
+    CreativeModeInstantBreak(){}
+}
+
+final class Flying implements PlayerAbility {
+    Flying(){}
+}
+
+final class Invulnerable implements PlayerAbility {
+    Invulnerable(){}
+}
diff --git a/src/main/java/com/github/weepingmc/packet/options/abilities/WalkSpeedAbility.java b/src/main/java/com/github/weepingmc/packet/options/abilities/WalkSpeedAbility.java
new file mode 100644
index 0000000000000000000000000000000000000000..8010fe740d38377aad68790c36afd5d69bb87569
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/packet/options/abilities/WalkSpeedAbility.java
@@ -0,0 +1,4 @@
+package com.github.weepingmc.packet.options.abilities;
+
+public record WalkSpeedAbility(float speed) implements PlayerAbility{
+}
