From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 9 Dec 2023 00:18:22 +0100
Subject: [PATCH] fix failing class reference


diff --git a/src/main/java/io/papermc/paper/registry/RegistryKey.java b/src/main/java/io/papermc/paper/registry/RegistryKey.java
index c4b30b16ce4db754b958c493ad86d0863592c263..3e39d859b1789ab7c5ab24121fe2f7a09d6a5f1e 100644
--- a/src/main/java/io/papermc/paper/registry/RegistryKey.java
+++ b/src/main/java/io/papermc/paper/registry/RegistryKey.java
@@ -32,12 +32,10 @@ public sealed interface RegistryKey<T> extends Keyed permits RegistryKeyImpl {
      * ******************* */
     /**
      * Built-in registry for game events
-     * @see io.papermc.paper.registry.keys.GameEventKeys
      */
     RegistryKey<GameEvent> GAME_EVENT = create("game_event");
     /**
      * Built-in registry for structure types.
-     * @see io.papermc.paper.registry.keys.StructureTypeKeys
      */
     RegistryKey<StructureType> STRUCTURE_TYPE = create("worldgen/structure_type");
 
@@ -46,22 +44,18 @@ public sealed interface RegistryKey<T> extends Keyed permits RegistryKeyImpl {
      * ********************** */
     /**
      * Data-driven registry for biomes.
-     * @see io.papermc.paper.registry.keys.BiomeKeys
      */
     RegistryKey<Biome> BIOME = create("worldgen/biome");
     /**
      * Data-driven registry for structures.
-     * @see io.papermc.paper.registry.keys.StructureKeys
      */
     RegistryKey<Structure> STRUCTURE = create("worldgen/structure");
     /**
      * Data-driven registry for trim materials.
-     * @see io.papermc.paper.registry.keys.TrimMaterialKeys
      */
     RegistryKey<TrimMaterial> TRIM_MATERIAL = create("trim_material");
     /**
      * Data-driven registry for trim patterns.
-     * @see io.papermc.paper.registry.keys.TrimPatternKeys
      */
     RegistryKey<TrimPattern> TRIM_PATTERN = create("trim_pattern");
 }
