From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Wed, 31 Jan 2024 21:42:20 +0100
Subject: [PATCH] add posibility to trigger reconfiguration


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index c6cb4f17469a8f2e60dd3e28d41402851ce5fb21..3b3c2adb78414bc00d35f6c38febfd6981149982 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3711,4 +3711,10 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    // Weeper start - trigger reconfiguration
+    /**
+     * Triggers Configuration phase
+     */
+    void triggerReconfiguration();
+    // Weeper end - trigger reconfiguration
 }
