From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Wed, 31 Jan 2024 21:42:20 +0100
Subject: [PATCH] add posibility to trigger reconfiguration


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 31aa6ef5fc2b3b88c72f5a15b8cc7a0e50c29f46..4c41aff3eec33c3a0ee0ec2429b5b173f0304101 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3619,4 +3619,10 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    // Weeper start - trigger reconfiguration
+    /**
+     * Triggers Configuration phase
+     */
+    void triggerReconfiguration();
+    // Weeper end - trigger reconfiguration
 }
