From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Wed, 31 Jan 2024 21:42:20 +0100
Subject: [PATCH] add posibility to trigger reconfiguration


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 254a02ddb5dc867c9dd6c2086791f7ab94247fd3..635f7ba441832af7e7db10f6276b84accee0f174 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3687,4 +3687,10 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    // Weeper start - trigger reconfiguration
+    /**
+     * Triggers Configuration phase
+     */
+    void triggerReconfiguration();
+    // Weeper end - trigger reconfiguration
 }
