From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 16 Mar 2024 22:58:19 +0100
Subject: [PATCH] add disguise api


diff --git a/src/main/java/com/destroystokyo/paper/PaperSkinParts.java b/src/main/java/com/destroystokyo/paper/PaperSkinParts.java
index b6f4400df3d8ec7e06a996de54f8cabba57885e1..ce28a4e9597b8c7a2d535eaac34d8babd67ed226 100644
--- a/src/main/java/com/destroystokyo/paper/PaperSkinParts.java
+++ b/src/main/java/com/destroystokyo/paper/PaperSkinParts.java
@@ -71,4 +71,59 @@ public class PaperSkinParts implements SkinParts {
             .add("hats=" + hasHatsEnabled())
             .toString();
     }
+
+    public static SkinParts.Builder builder(){
+        return new Builder();
+    }
+
+    public static class Builder implements SkinParts.Builder{
+        private int raw = 0;
+
+        private static final int CAPE = 0x01;
+        private static final int JACKET = 0x02;
+        private static final int LEFT_SLEEVE = 0x04;
+        private static final int RIGHT_SLEEVE = 0x08;
+        private static final int LEFT_PANTS = 0x10;
+        private static final int RIGHT_PANTS = 0x20;
+        private static final int HAT = 0x40;
+
+        public @org.jetbrains.annotations.NotNull Builder withCape(){
+            raw |= CAPE;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull Builder withJacket(){
+            raw |= JACKET;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull Builder withLeftSleeve(){
+            raw |= LEFT_SLEEVE;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull Builder withRightSleeve(){
+            raw |= RIGHT_SLEEVE;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull Builder withLeftPants(){
+            raw |= LEFT_PANTS;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull Builder withRightPants(){
+            raw |= RIGHT_PANTS;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull Builder withHat(){
+            raw |= HAT;
+            return this;
+        }
+
+        public @org.jetbrains.annotations.NotNull SkinParts build(){
+            return new PaperSkinParts(raw);
+        }
+    }
 }
diff --git a/src/main/java/com/github/weepingmc/disguise/DisguiseUtil.java b/src/main/java/com/github/weepingmc/disguise/DisguiseUtil.java
new file mode 100644
index 0000000000000000000000000000000000000000..e5c8fac57435a2f955e4de1061e8987933c71c4d
--- /dev/null
+++ b/src/main/java/com/github/weepingmc/disguise/DisguiseUtil.java
@@ -0,0 +1,94 @@
+package com.github.weepingmc.disguise;
+
+import com.destroystokyo.paper.profile.PlayerProfile;
+import java.util.EnumSet;
+import java.util.List;
+import net.minecraft.network.protocol.game.ClientboundAddEntityPacket;
+import net.minecraft.network.syncher.EntityDataAccessor;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.phys.Vec3;
+import org.bukkit.Bukkit;
+
+public final class DisguiseUtil {
+    private DisguiseUtil(){}
+
+    public static boolean tryDisguise(ServerPlayer player, Entity entity, ClientboundAddEntityPacket clientboundAddEntityPacket) {
+            switch (entity.getBukkitEntity().getDisguiseData()) {
+                case DisguiseData.OriginalDisguise disguise -> {
+                    return false;
+                }
+                case com.github.weepingmc.disguise.EntityTypeDisguise(var type) -> {
+                    player.connection.send(new net.minecraft.network.protocol.game.ClientboundAddEntityPacket(
+                        clientboundAddEntityPacket.getId(),
+                        clientboundAddEntityPacket.getUUID(),
+                        clientboundAddEntityPacket.getX(),
+                        clientboundAddEntityPacket.getY(),
+                        clientboundAddEntityPacket.getZ(),
+                        clientboundAddEntityPacket.getXRot(),
+                        clientboundAddEntityPacket.getYRot(),
+                        org.bukkit.craftbukkit.entity.CraftEntityType.bukkitToMinecraft(type),
+                        0,
+                        Vec3.ZERO.add(clientboundAddEntityPacket.getX(), clientboundAddEntityPacket.getY(), clientboundAddEntityPacket.getZ()).scale(1/8000.0D),
+                        clientboundAddEntityPacket.getYHeadRot()
+                    ));
+
+                    return true;
+                }
+                case com.github.weepingmc.disguise.PlayerDisguise(var playerProfile, var listed, var skinParts) -> {
+                    PlayerProfile adapted = Bukkit.createProfile(entity.getUUID(), entity.hasCustomName()? entity.getBukkitEntity().getCustomName(): playerProfile.getName());
+                    adapted.setProperties(playerProfile.getProperties());
+                    net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Entry playerUpdate =
+                        new net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Entry(
+                            entity.getUUID(),
+                            com.destroystokyo.paper.profile.CraftPlayerProfile.asAuthlibCopy(adapted),
+                            listed,
+                            0,
+                            net.minecraft.world.level.GameType.DEFAULT_MODE,
+                            entity.getCustomName(),
+                            null
+                        );
+                    player.connection.send(new net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket(EnumSet.of(net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Action.ADD_PLAYER), playerUpdate));
+
+                    player.connection.send(new net.minecraft.network.protocol.game.ClientboundAddEntityPacket(
+                        clientboundAddEntityPacket.getId(),
+                        clientboundAddEntityPacket.getUUID(),
+                        clientboundAddEntityPacket.getX(),
+                        clientboundAddEntityPacket.getY(),
+                        clientboundAddEntityPacket.getZ(),
+                        clientboundAddEntityPacket.getXRot(),
+                        clientboundAddEntityPacket.getYRot(),
+                        net.minecraft.world.entity.EntityType.PLAYER,
+                        0,
+                        Vec3.ZERO.add(clientboundAddEntityPacket.getX(), clientboundAddEntityPacket.getY(), clientboundAddEntityPacket.getZ()).scale(1/8000.0D),
+                        clientboundAddEntityPacket.getYHeadRot()
+                    ));
+                    if(skinParts != null) {
+                        player.connection.send(new net.minecraft.network.protocol.game.ClientboundSetEntityDataPacket(
+                            clientboundAddEntityPacket.getId(),
+                            List.of(new SynchedEntityData.DataItem<>(Player.DATA_PLAYER_MODE_CUSTOMISATION, (byte) skinParts.getRaw()).value())
+                        ));
+                    }
+                    return true;
+                }
+            }
+        }
+
+    public static boolean shouldSkip(Entity entity, EntityDataAccessor<?> dataAccessor) {
+        return switch (entity.getBukkitEntity().getDisguiseData()) {
+            case DisguiseData.OriginalDisguise original -> {
+                yield false;
+            }
+            case EntityTypeDisguise entityTypeDisguise -> !io.papermc.paper.entity.meta.EntityMetaWatcher.isValidForClass(
+                org.bukkit.craftbukkit.entity.CraftEntityType.bukkitToMinecraft(entityTypeDisguise.entityType()).getBaseClass(),
+                dataAccessor
+            );
+            case PlayerDisguise playerDisguise -> !io.papermc.paper.entity.meta.EntityMetaWatcher.isValidForClass(
+                ServerPlayer.class,
+                dataAccessor
+            );
+        };
+    }
+}
diff --git a/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java b/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java
index 0f99733660f91280e4c6262cf75b3c9cae86f65a..439211af09b50cd05ce99b6285ce39336ae7825b 100644
--- a/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java
@@ -100,6 +100,7 @@ public class SynchedEntityData {
 
                 if (datawatcher_item.isDirty()) {
                     datawatcher_item.setDirty(false);
+                    if (com.github.weepingmc.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, datawatcher_item.getAccessor())) continue; // Weeper - disguise api
                     list.add(datawatcher_item.value());
                 }
             }
@@ -116,7 +117,7 @@ public class SynchedEntityData {
 
         for (int j = 0; j < i; ++j) {
             SynchedEntityData.DataItem<?> datawatcher_item = adatawatcher_item[j];
-
+            if (com.github.weepingmc.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, datawatcher_item.getAccessor())) continue; // Weeper - disguise api
             if (!datawatcher_item.isSetToDefault()) {
                 if (list == null) {
                     list = new ArrayList();
@@ -135,7 +136,7 @@ public class SynchedEntityData {
         while (iterator.hasNext()) {
             SynchedEntityData.DataValue<?> datawatcher_c = (SynchedEntityData.DataValue) iterator.next();
             SynchedEntityData.DataItem<?> datawatcher_item = this.itemsById[datawatcher_c.id];
-
+            if (com.github.weepingmc.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, datawatcher_item.getAccessor())) continue; // Weeper - disguise api
             this.assignValue(datawatcher_item, datawatcher_c);
             this.entity.onSyncedDataUpdated(datawatcher_item.getAccessor());
         }
@@ -158,6 +159,7 @@ public class SynchedEntityData {
     public List<SynchedEntityData.DataValue<?>> packAll() {
         final List<SynchedEntityData.DataValue<?>> list = new ArrayList<>();
         for (final DataItem<?> dataItem : this.itemsById) {
+            if (com.github.weepingmc.disguise.DisguiseUtil.shouldSkip((net.minecraft.world.entity.Entity) entity, dataItem.getAccessor())) continue; // Weeper - disguise api
             list.add(dataItem.value());
         }
 
diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index f16a69775332a08ed0e87d27acd0fc959359694c..0d4adf1e67597348a5592a539d93390c8b7254b3 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -316,7 +316,11 @@ public class ServerEntity {
         Packet<ClientGamePacketListener> packet = this.entity.getAddEntityPacket();
 
         this.yHeadRotp = Mth.floor(this.entity.getYHeadRot() * 256.0F / 360.0F);
-        sender.accept(packet);
+        // Weeper start - disguise api
+        if(!com.github.weepingmc.disguise.DisguiseUtil.tryDisguise(player, entity, (net.minecraft.network.protocol.game.ClientboundAddEntityPacket) packet)){
+            sender.accept(packet);
+        }
+        // Weeper end - disguise api
         if (this.trackedDataValues != null) {
             sender.accept(new ClientboundSetEntityDataPacket(this.entity.getId(), this.trackedDataValues));
         }
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 8aff6ffb6f25c90cfb21ce9c710fc0f55b4b17c9..4fb1253177114511d2fbc7255756bade36ec6959 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -380,6 +380,12 @@ public final class CraftServer implements Server {
         return io.papermc.paper.util.TickThread.isTickThreadFor(((org.bukkit.craftbukkit.entity.CraftEntity) entity).getHandleRaw());
     }
     // Paper end - Folia reagion threading API
+    // Weeper start - add disguise api
+    @Override
+    public com.destroystokyo.paper.SkinParts.@org.jetbrains.annotations.NotNull Builder newSkinPartsBuilder() {
+        return com.destroystokyo.paper.PaperSkinParts.builder();
+    }
+    // Weeper end - add disguise api
 
     static {
         ConfigurationSerialization.registerClass(CraftOfflinePlayer.class);
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index a2d336ceb52b63db5c03432ee7bc94dc6a742b82..3c60bd79e2b41785f65b0714b7d6b8fd76009165 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -1280,4 +1280,16 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         return this.getHandle().getScoreboardName();
     }
     // Paper end - entity scoreboard name
+    // Weeper start - disguise api
+    private com.github.weepingmc.disguise.DisguiseData disguiseData = com.github.weepingmc.disguise.DisguiseData.original();
+    @Override
+    public @org.jetbrains.annotations.NotNull com.github.weepingmc.disguise.DisguiseData getDisguiseData() {
+        return disguiseData;
+    }
+
+    @Override
+    public void setDisuiseData(@org.jetbrains.annotations.NotNull com.github.weepingmc.disguise.DisguiseData disguiseData) {
+        this.disguiseData = disguiseData;
+    }
+    // Weeper end - disguise api
 }
