From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:08:28 +0200
Subject: [PATCH] trigger Block animation


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 1b6ae90acffa06502902e11473b65c8431616b05..b4edbb921f48d36cf5d52c0209b90756f3a00193 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -55,6 +55,7 @@ import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.ClipContext;
 import net.minecraft.world.level.Explosion;
 import net.minecraft.world.level.GameRules;
+import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.chunk.ChunkAccess;
 import net.minecraft.world.level.chunk.ChunkStatus;
 import net.minecraft.world.level.chunk.ImposterProtoChunk;
@@ -135,6 +136,7 @@ import org.bukkit.util.Consumer;
 import org.bukkit.util.RayTraceResult;
 import org.bukkit.util.StructureSearchResult;
 import org.bukkit.util.Vector;
+import org.jetbrains.annotations.NotNull;
 
 public class CraftWorld extends CraftRegionAccessor implements World {
     public static final int CUSTOM_DIMENSION_OFFSET = 10;
@@ -2247,6 +2249,17 @@ public class CraftWorld extends CraftRegionAccessor implements World {
         return (this.getHandle().dragonFight() == null) ? null : new CraftDragonBattle(this.getHandle().dragonFight());
     }
 
+
+    // Weeper start
+    @Override
+    public void triggerBlockAnimation(@NotNull Location location,
+                                      int action, int actionParameter) {
+        BlockPos position = new BlockPos(location.getX(), location.getY(), location.getZ());
+        BlockEntity blockEntity = world.getBlockEntity(position, true);
+        world.blockEvent(position, blockEntity.getBlockState().getBlock(), action, actionParameter);
+    }
+    // Weeper end
+
     @Override
     public PersistentDataContainer getPersistentDataContainer() {
         return this.persistentDataContainer;
