From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:46:18 +0200
Subject: [PATCH] player track entity events


diff --git a/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java b/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
index 1ed97a5e2b5b77d828113be9c0ea178de99eaab5..6de1ee8d28d08ff623f4ea1a1595398d569e39b0 100644
--- a/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
+++ b/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
@@ -34,7 +34,7 @@ public final class DisguiseUtil {
 
     public static boolean tryDisguise(ServerPlayer player, Entity entity, Packet<?> packet) {
             if(!(packet instanceof ClientboundAddEntityPacket clientboundAddEntityPacket)) {
-                return false;
+                return !(com.github.weepingmc.event.packet.PlayerTrackEntityEvent.getHandlerList().getRegisteredListeners().length == 0 || new com.github.weepingmc.event.packet.PlayerTrackEntityEvent(player.getBukkitEntity(), entity.getBukkitEntity()).callEvent()); // Weeper - player track entity events
             }
             return switch (entity.getBukkitEntity().getDisguiseData()) {
                 case DisguiseData.OriginalDisguise disguise -> false;
@@ -97,6 +97,7 @@ public final class DisguiseUtil {
         );
     }
 
+    private static final org.bukkit.NamespacedKey skinKey = new org.bukkit.NamespacedKey("npclib", "npclib-skin"); // Weeper - track entity events
 
     /*
      * Is used to skip entity meta that doesn't fit the disguised type.
@@ -109,6 +110,11 @@ public final class DisguiseUtil {
     }
 
     public static boolean shouldSkip(Entity entity, EntityDataSerializer<?> entityDataSerializer, int id) {
+        // Weeper start - track entity events
+        if (entity.getBukkitEntity().getPersistentDataContainer().has(skinKey, org.bukkit.persistence.PersistentDataType.STRING)) {
+            return !io.papermc.paper.entity.meta.EntityMetaWatcher.isValidForClass(ServerPlayer.class, entityDataSerializer, id);
+        }
+        // Weeper end - track entity events
         return switch (entity.getBukkitEntity().getDisguiseData()) {
             case DisguiseData.OriginalDisguise original -> false;
             case EntityTypeDisguise entityTypeDisguise -> !io.papermc.paper.entity.meta.EntityMetaWatcher.isValidForClass(
diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index ea7527e1e5399172ac9bed70ee51306e1c47fd33..6404e32979e2206bfd97db51b1e408cc24eb3b5e 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -336,6 +336,7 @@ public class ServerEntity {
         this.entity.stopSeenByPlayer(player);
         io.papermc.paper.disguise.DisguiseUtil.tryDespawn(player, this.entity); // Paper - disguise api
         player.connection.send(new ClientboundRemoveEntitiesPacket(new int[]{this.entity.getId()}));
+        if(com.github.weepingmc.event.packet.PlayerUntrackEntityEvent.getHandlerList().getRegisteredListeners().length != 0) new com.github.weepingmc.event.packet.PlayerUntrackEntityEvent(player.getBukkitEntity(), this.entity.getId(), this.entity.getBukkitEntity()).callEvent(); // Weeper - player track entity events
     }
 
     public void addPairing(ServerPlayer player) {
