From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 22:31:23 +0200
Subject: [PATCH] add mechanix to generate comments to vanilla mob goals


diff --git a/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java b/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java
index 654ed436e99dd56f1fe7c1d4f38da34d95ce9349..2db270a39515af0525a0d82b508627d26285e930 100644
--- a/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java
+++ b/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java
@@ -57,6 +57,11 @@ public class VanillaMobGoalTest {
         if (missingFromAPI.size() != 0) {
             System.out.println("Missing from API: ");
             for (GoalKey<?> key : missingFromAPI) {
+                System.out.printf("""
+                        /**
+                        * MobGoal %s applicable to any entity that is instance of %s
+                        */
+                        """, key.getNamespacedKey().getKey(), key.getEntityClass().getSimpleName());
                 System.out.println("GoalKey<" + key.getEntityClass().getSimpleName() + "> " + key.getNamespacedKey().getKey().toUpperCase() +
                     " = GoalKey.of(" + key.getEntityClass().getSimpleName() + ".class, NamespacedKey.minecraft(\"" + key.getNamespacedKey().getKey() + "\"));");
             }
