From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 19 Jun 2021 15:46:18 +0200
Subject: [PATCH] player track entity events


diff --git a/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java b/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
index f591aec5a8525a7f1c3c329d357f0bb0b199c0be..0231747c436bacdd103afb86f976c0bc652431d3 100644
--- a/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
+++ b/src/main/java/io/papermc/paper/disguise/DisguiseUtil.java
@@ -26,7 +26,7 @@ public final class DisguiseUtil {
 
     public static boolean tryDisguise(ServerPlayer player, Entity entity, Packet<?> packet) {
             if(!(packet instanceof ClientboundAddEntityPacket clientboundAddEntityPacket)) {
-                return false;
+                return !(com.github.weepingmc.event.packet.PlayerTrackEntityEvent.getHandlerList().getRegisteredListeners().length == 0 || new com.github.weepingmc.event.packet.PlayerTrackEntityEvent(player.getBukkitEntity(), entity.getBukkitEntity()).callEvent()); // Weeper - player track entity events
             }
             return switch (entity.getBukkitEntity().getDisguiseData()) {
                 case DisguiseData.OriginalDisguise disguise -> false;
@@ -88,6 +88,7 @@ public final class DisguiseUtil {
         );
     }
 
+    private static final org.bukkit.NamespacedKey skinKey = new org.bukkit.NamespacedKey("npclib", "npclib-skin"); // Weeper - track entity events
 
     /*
      * Is used to skip entity meta that doesn't fit the disguised type.
@@ -95,6 +96,11 @@ public final class DisguiseUtil {
      * that has a byte at that index.
      */
     public static boolean shouldSkip(Entity entity, EntityDataAccessor<?> dataAccessor) {
+        // Weeper start - track entity events
+        if (entity.getBukkitEntity().getPersistentDataContainer().has(skinKey, org.bukkit.persistence.PersistentDataType.STRING)) {
+            return !io.papermc.paper.entity.meta.EntityMetaWatcher.isValidForClass(ServerPlayer.class, dataAccessor);
+        }
+        // Weeper end - track entity events
         return switch (entity.getBukkitEntity().getDisguiseData()) {
             case DisguiseData.OriginalDisguise original -> false;
             case EntityTypeDisguise entityTypeDisguise -> !io.papermc.paper.entity.meta.EntityMetaWatcher.isValidForClass(
diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index 4c9615d7c72598e57a138e2fae5530eccf6ef292..d057a47e3b2f22e1440946fc2c0eb7eaa57693a1 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -328,6 +328,7 @@ public class ServerEntity {
         this.entity.stopSeenByPlayer(player);
         io.papermc.paper.disguise.DisguiseUtil.tryDespawn(player, this.entity); // Paper - disguise api
         player.connection.send(new ClientboundRemoveEntitiesPacket(new int[]{this.entity.getId()}));
+        if(com.github.weepingmc.event.packet.PlayerUntrackEntityEvent.getHandlerList().getRegisteredListeners().length != 0) new com.github.weepingmc.event.packet.PlayerUntrackEntityEvent(player.getBukkitEntity(), this.entity.getId(), this.entity.getBukkitEntity()).callEvent(); // Weeper - player track entity events
     }
 
     public void addPairing(ServerPlayer player) {
