From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Thu, 8 Jul 2021 00:11:30 +0200
Subject: [PATCH] temporarily get rid of reobfuscation definition in build
 gradle


diff --git a/build.gradle.kts b/build.gradle.kts
index dd579110fa57d7cbe4a85c8dfd5fc14ab186c3ee..e2a17bab86f378d382839c8b8585dab94bc18c30 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,21 +1,14 @@
 import com.github.jengelman.gradle.plugins.shadow.transformers.Log4j2PluginsCacheFileTransformer
 import com.github.jengelman.gradle.plugins.shadow.transformers.Transformer
 import io.papermc.paperweight.tasks.BaseTask
-import io.papermc.paperweight.tasks.GenerateReobfMappings
-import io.papermc.paperweight.tasks.PatchMappings
-import io.papermc.paperweight.util.Constants
 import io.papermc.paperweight.util.Git
-import io.papermc.paperweight.util.cache
 import io.papermc.paperweight.util.defaultOutput
 import io.papermc.paperweight.util.openZip
 import io.papermc.paperweight.util.path
-import io.papermc.paperweight.util.registering
-import io.papermc.paperweight.util.set
 import shadow.org.apache.logging.log4j.core.config.plugins.processor.PluginProcessor.PLUGIN_CACHE_FILE
 import java.nio.file.Files
 import java.text.SimpleDateFormat
-import java.util.Date
-import java.util.Locale
+import java.util.*
 
 plugins {
     java
@@ -113,22 +106,6 @@ publishing {
     }
 }
 
-val generateReobfMappings = rootProject.tasks.named<GenerateReobfMappings>("generateReobfMappings")
-
-val patchReobfMappings by tasks.registering<PatchMappings> {
-    inputMappings.set(generateReobfMappings.flatMap { it.reobfMappings })
-    patch.set(rootProject.layout.cache.resolve("paperweight/upstreams/paper/build-data/reobf-mappings-patch.tiny"))
-
-    fromNamespace.set(Constants.DEOBF_NAMESPACE)
-    toNamespace.set(Constants.SPIGOT_NAMESPACE)
-
-    outputMappings.set(layout.cache.resolve("paperweight/mappings/reobf-patched.tiny"))
-}
-
-tasks.reobfJar {
-    mappingsFile.set(patchReobfMappings.flatMap { it.outputMappings })
-}
-
 val generatePom = tasks.named<GenerateMavenPom>("generatePomFileForMavenPublication")
 
 tasks.shadowJar {
