From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Thu, 8 Jul 2021 23:23:01 +0200
Subject: [PATCH] add marker support


diff --git a/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java b/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java
index 175f3a2a3d1d87c065707911d64a669ff0be5729..c8f54a65f9a42d658c055c2648d924724098fce6 100644
--- a/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java
+++ b/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java
@@ -76,7 +76,19 @@ public class CraftPacketStepBuilder implements PacketStepBuilder {
     @Override
     @Nonnull
     public PacketStepBuilder spawnPlayer(int entityId, @Nonnull UUID uuid, @Nonnull Location location) {
-        initial.setNext(of(new ClientboundAddPlayerPacket(entityId, uuid, location)));
+        initial.setNext(of(new ClientboundAddEntityPacket(
+            entityId,
+            uuid,
+            location.x(),
+            location.y(),
+            location.z(),
+            location.getPitch(),
+            location.getYaw(),
+            net.minecraft.world.entity.EntityType.PLAYER,
+            0,
+            Vec3.ZERO,
+            location.getYaw()
+        )));
         return this;
     }
 
@@ -255,6 +267,18 @@ public class CraftPacketStepBuilder implements PacketStepBuilder {
         return this;
     }
 
+    @Override
+    public @NotNull PacketStepBuilder showTestMarker(@NotNull Location location, @NotNull Color color, @org.jetbrains.annotations.Nullable String text, int time) {
+        var payload = new net.minecraft.network.protocol.common.custom.GameTestAddMarkerDebugPayload(
+            new BlockPos(location.getBlockX(), location.getBlockY(), location.getBlockZ()),
+            org.bukkit.Color.fromARGB(color.getAlpha(), color.getRed(), color.getGreen(), color.getBlue()).asARGB(),
+            text==null?"": text,
+            time
+        );
+        initial.setNext(of(new net.minecraft.network.protocol.common.ClientboundCustomPayloadPacket(payload)));
+        return this;
+    }
+
     @Override
     public void send(@Nonnull Collection<? extends org.bukkit.entity.Player> players) {
         initial.execute(players);
