From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 6 Nov 2021 23:33:15 +0100
Subject: [PATCH] added executor for easy usage with CompletableFutures


diff --git a/src/main/java/org/bukkit/craftbukkit/scheduler/MinecraftInternalPlugin.java b/src/main/java/org/bukkit/craftbukkit/scheduler/MinecraftInternalPlugin.java
index 909b2c98e7a9117d2f737245e4661792ffafb744..d320fd1ec4fbdd66c68e505db000d253cff32131 100644
--- a/src/main/java/org/bukkit/craftbukkit/scheduler/MinecraftInternalPlugin.java
+++ b/src/main/java/org/bukkit/craftbukkit/scheduler/MinecraftInternalPlugin.java
@@ -11,12 +11,14 @@ import org.bukkit.plugin.PluginBase;
 import org.bukkit.plugin.PluginDescriptionFile;
 import org.bukkit.plugin.PluginLoader;
 import org.bukkit.plugin.PluginLogger;
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.io.File;
 import java.io.InputStream;
 import java.util.List;
+import java.util.concurrent.Executor;
 
 public class MinecraftInternalPlugin extends PluginBase {
     private boolean enabled = true;
@@ -88,6 +90,16 @@ public class MinecraftInternalPlugin extends PluginBase {
         throw new UnsupportedOperationException("Not supported.");
     }
 
+    @Override
+    public @NotNull Executor mainThreadExecutor() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    @Override
+    public @NonNull Executor asyncExecutor() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
     @Override
     public boolean isEnabled() {
         return enabled;
