From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 5 Mar 2022 17:27:38 +0100
Subject: [PATCH] do not check if registry is frozen


diff --git a/src/main/java/net/minecraft/core/MappedRegistry.java b/src/main/java/net/minecraft/core/MappedRegistry.java
index b2d22bb25499756fe393cf562f8a4eda0649c26d..5b99b1f144b55a85979999104e47ce5e50436a51 100644
--- a/src/main/java/net/minecraft/core/MappedRegistry.java
+++ b/src/main/java/net/minecraft/core/MappedRegistry.java
@@ -126,7 +126,7 @@ public class MappedRegistry<T> implements WritableRegistry<T> {
     }
 
     public Holder.Reference<T> registerMapping(int rawId, ResourceKey<T> key, T value, Lifecycle lifecycle) {
-        this.validateWrite(key);
+        // this.validateWrite(key); // Weepeer do not validate freeze
         Validate.notNull(key);
         Validate.notNull(value);
         if (this.byLocation.containsKey(key.location())) {
