From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 5 Mar 2022 17:27:38 +0100
Subject: [PATCH] do not check if registry is frozen


diff --git a/src/main/java/net/minecraft/core/MappedRegistry.java b/src/main/java/net/minecraft/core/MappedRegistry.java
index 3ef5d440fd3ce209110543bfd36569e846a5b749..acb18e1e04aba21d817eec14d4e7cc5681f20135 100644
--- a/src/main/java/net/minecraft/core/MappedRegistry.java
+++ b/src/main/java/net/minecraft/core/MappedRegistry.java
@@ -84,7 +84,7 @@ public class MappedRegistry<T> extends WritableRegistry<T> {
     }
 
     private Holder<T> registerMapping(int rawId, ResourceKey<T> key, T value, Lifecycle lifecycle, boolean checkDuplicateKeys) {
-        this.validateWrite(key);
+        // this.validateWrite(key); // Weepeer do not validate freeze
         Validate.notNull(key);
         Validate.notNull(value);
         this.byId.size(Math.max(this.byId.size(), rawId + 1));
