From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 27 Nov 2021 14:47:17 +0100
Subject: [PATCH] removed double declaration of storage and palette in record


diff --git a/src/main/java/net/minecraft/world/level/chunk/PalettedContainer.java b/src/main/java/net/minecraft/world/level/chunk/PalettedContainer.java
index 23cb543ff5e82e2a6e5b3a32ce776907bb0598d2..7f8445c3fbd8a27d19512c38ca90133209a9ae65 100644
--- a/src/main/java/net/minecraft/world/level/chunk/PalettedContainer.java
+++ b/src/main/java/net/minecraft/world/level/chunk/PalettedContainer.java
@@ -300,8 +300,6 @@ public class PalettedContainer<T> implements PaletteResize<T> {
     }
 
     static record Data<T>(PalettedContainer.Configuration<T> configuration, BitStorage storage, Palette<T> palette) {
-        final BitStorage storage;
-        final Palette<T> palette;
 
         public void copyFrom(Palette<T> palette, BitStorage storage) {
             for(int i = 0; i < storage.getSize(); ++i) {
