From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Sat, 5 Mar 2022 17:27:38 +0100
Subject: [PATCH] do not check if registry is frozen


diff --git a/src/main/java/net/minecraft/core/MappedRegistry.java b/src/main/java/net/minecraft/core/MappedRegistry.java
index 1e33434f9f361542e03da3e4812bc6d76768a202..9aa011364128094dda5ab13ad55d444b80a08e12 100644
--- a/src/main/java/net/minecraft/core/MappedRegistry.java
+++ b/src/main/java/net/minecraft/core/MappedRegistry.java
@@ -130,7 +130,7 @@ public class MappedRegistry<T> implements WritableRegistry<T> {
 
     @Override
     public Holder.Reference<T> registerMapping(int i, ResourceKey<T> resourceKey, T object, Lifecycle lifecycle) {
-        this.validateWrite(resourceKey);
+        // this.validateWrite(resourceKey); // Weepeer do not validate freeze
         Validate.notNull(resourceKey);
         Validate.notNull(object);
         if (this.byLocation.containsKey(resourceKey.location())) {
