From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Mon, 16 Oct 2023 19:48:03 +0200
Subject: [PATCH] modern player abilities


diff --git a/src/main/java/com/github/weepingmc/packet/CraftPacketConversion.java b/src/main/java/com/github/weepingmc/packet/CraftPacketConversion.java
index c7e3556894aed7d09d8323155d9467a2a7930262..1fd861001d19ab3c14b6239764d53621b0b07c32 100644
--- a/src/main/java/com/github/weepingmc/packet/CraftPacketConversion.java
+++ b/src/main/java/com/github/weepingmc/packet/CraftPacketConversion.java
@@ -80,7 +80,7 @@ public final class CraftPacketConversion {
         };
     }
 
-    static Abilities mapPlayerAbilities(Set<PlayerAbility> playerAbilities) {
+    static Abilities mapPlayerAbilitiesOld(Set<PlayerAbility> playerAbilities) {
         Abilities playerAbilitiesNMS = new Abilities();
 
         for (PlayerAbility playerAbilitiy : playerAbilities) {
@@ -93,4 +93,26 @@ public final class CraftPacketConversion {
         }
         return playerAbilitiesNMS;
     }
+
+    static Abilities mapPlayerAbilities(Set<com.github.weepingmc.packet.options.abilities.PlayerAbility> playerAbilities) {
+        Abilities playerAbilitiesNMS = new Abilities();
+        for (com.github.weepingmc.packet.options.abilities.PlayerAbility playerAbilitiy : playerAbilities) {
+            if (playerAbilitiy instanceof com.github.weepingmc.packet.options.abilities.FlySpeedAbility(float speed)) {
+                playerAbilitiesNMS.flyingSpeed = speed;
+            } else if (playerAbilitiy instanceof com.github.weepingmc.packet.options.abilities.WalkSpeedAbility(float speed)) {
+                playerAbilitiesNMS.walkingSpeed = speed;
+            } else if (playerAbilitiy == com.github.weepingmc.packet.options.abilities.PlayerAbility.INVULNERABLE) {
+                playerAbilitiesNMS.invulnerable = true;
+            } else if (playerAbilitiy == com.github.weepingmc.packet.options.abilities.PlayerAbility.FLYING) {
+                playerAbilitiesNMS.flying = true;
+            } else if (playerAbilitiy == com.github.weepingmc.packet.options.abilities.PlayerAbility.ALLOW_FLYING) {
+                playerAbilitiesNMS.mayfly = true;
+            } else if (playerAbilitiy == com.github.weepingmc.packet.options.abilities.PlayerAbility.CREATIVE_MODE_INSTANT_BREAK) {
+                playerAbilitiesNMS.instabuild = true;
+            } else {
+                throw new IllegalStateException("Unexpected value: " + playerAbilitiy);
+            }
+        }
+        return playerAbilitiesNMS;
+    }
 }
diff --git a/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java b/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java
index d639773964fb9a6ca07cee58b7cb41807b9334ae..500ac16dd4a7b379dac0d45e90111a71e1027012 100644
--- a/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java
+++ b/src/main/java/com/github/weepingmc/packet/CraftPacketStepBuilder.java
@@ -268,14 +268,21 @@ public class CraftPacketStepBuilder implements PacketStepBuilder {
     }
 
     @Override
-    public PacketStepBuilder playerAbilities(@Nonnull Set<PlayerAbility> playerAbilities, float flySpeed, float fieldOfViewModifier) {
-        Abilities playerAbilitiesNMS = mapPlayerAbilities(playerAbilities);
+    public @NotNull PacketStepBuilder playerAbilities(@Nonnull Set<PlayerAbility> playerAbilities, float flySpeed, float fieldOfViewModifier) {
+        Abilities playerAbilitiesNMS = com.github.weepingmc.packet.CraftPacketConversion.mapPlayerAbilitiesOld(playerAbilities);
         playerAbilitiesNMS.flyingSpeed = flySpeed;
         playerAbilitiesNMS.walkingSpeed = fieldOfViewModifier;
         initial.setNext(of(new ClientboundPlayerAbilitiesPacket(playerAbilitiesNMS)));
         return this;
     }
 
+    @Override
+    public @NotNull PacketStepBuilder withPlayerAbilities(@NotNull Set<com.github.weepingmc.packet.options.abilities.PlayerAbility> playerAbilities) {
+        Abilities playerAbilitiesNMS = mapPlayerAbilities(playerAbilities);
+        initial.setNext(of(new ClientboundPlayerAbilitiesPacket(playerAbilitiesNMS)));
+        return this;
+    }
+
     @Override
     public @NotNull PacketStepBuilder showTestMarker(@NotNull Location location, @NotNull Color color, @Nullable String text, int time) {
         var payload = new GameTestAddMarkerDebugPayload(
