From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Tue, 21 Jun 2022 20:49:19 +0200
Subject: [PATCH] Make goals public, needs migration to at sometime


diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
index f6062bf8c888baeb7b421150a2c64bf1af1a312b..bb2d354e6eabb927e32fd372f9cd37b061e1c6c4 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
@@ -691,7 +691,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         return pos.closerThan(this.blockPosition(), (double) distance);
     }
 
-    private class BeePollinateGoal extends Bee.BaseBeeGoal {
+    public class BeePollinateGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         private static final int MIN_POLLINATION_TICKS = 400;
         private static final int MIN_FIND_FLOWER_RETRY_COOLDOWN = 20;
@@ -877,7 +877,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeLookControl extends LookControl {
+    public class BeeLookControl extends LookControl { // Weeper private -> public
 
         BeeLookControl(Mob entity) {
             super(entity);
@@ -913,7 +913,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeEnterHiveGoal extends Bee.BaseBeeGoal {
+    public class BeeEnterHiveGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         BeeEnterHiveGoal() {
             super();
@@ -958,7 +958,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeLocateHiveGoal extends Bee.BaseBeeGoal {
+    public class BeeLocateHiveGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         BeeLocateHiveGoal() {
             super();
@@ -1190,7 +1190,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeGrowCropGoal extends Bee.BaseBeeGoal {
+    public class BeeGrowCropGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         static final int GROW_CHANCE = 30;
 
@@ -1254,7 +1254,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeWanderGoal extends Goal {
+    public class BeeWanderGoal extends Goal { // Weeper private -> public
 
         private static final int WANDER_THRESHOLD = 22;
 
@@ -1301,7 +1301,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeHurtByOtherGoal extends HurtByTargetGoal {
+    public class BeeHurtByOtherGoal extends HurtByTargetGoal { // Weeper private -> public
 
         BeeHurtByOtherGoal(Bee entitybee) {
             super(entitybee);
@@ -1321,7 +1321,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private static class BeeBecomeAngryTargetGoal extends NearestAttackableTargetGoal<Player> {
+    public static class BeeBecomeAngryTargetGoal extends NearestAttackableTargetGoal<Player> { // Weeper private -> public
 
         BeeBecomeAngryTargetGoal(Bee bee) {
             // Objects.requireNonNull(entitybee); // CraftBukkit - decompile error
@@ -1352,7 +1352,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private abstract class BaseBeeGoal extends Goal {
+    public abstract class BaseBeeGoal extends Goal { // Weeper private -> public
 
         BaseBeeGoal() {}
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Cat.java b/src/main/java/net/minecraft/world/entity/animal/Cat.java
index 90ce201bc7c47cef9bc59d7b535a7453854bac75..3858d2d3f1ac0481c5a9d17bda7754b04d6cf416 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Cat.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Cat.java
@@ -506,7 +506,7 @@ public class Cat extends TamableAnimal implements VariantHolder<CatVariant> {
         }
     }
 
-    private static class CatRelaxOnOwnerGoal extends Goal {
+    public static class CatRelaxOnOwnerGoal extends Goal { // Weeper private -> public
 
         private final Cat cat;
         @Nullable
@@ -652,7 +652,7 @@ public class Cat extends TamableAnimal implements VariantHolder<CatVariant> {
         }
     }
 
-    private static class CatAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
+    public static class CatAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> { // Weeper private -> public
 
         private final Cat cat;
 
diff --git a/src/main/java/net/minecraft/world/entity/raid/Raider.java b/src/main/java/net/minecraft/world/entity/raid/Raider.java
index 57fdcdaf54fd1c92a6e51a3a81789029096e5abe..bfe4ba5a5b982ccc88c5c89807fdcce97b59b04a 100644
--- a/src/main/java/net/minecraft/world/entity/raid/Raider.java
+++ b/src/main/java/net/minecraft/world/entity/raid/Raider.java
@@ -352,7 +352,7 @@ public abstract class Raider extends PatrollingMonster {
         }
     }
 
-    private static class RaiderMoveThroughVillageGoal extends Goal {
+    public static class RaiderMoveThroughVillageGoal extends Goal { // Weeper private -> public
 
         private final Raider raider;
         private final double speedModifier;
@@ -500,7 +500,7 @@ public abstract class Raider extends PatrollingMonster {
         }
     }
 
-    protected class HoldGroundAttackGoal extends Goal {
+    public class HoldGroundAttackGoal extends Goal { // Weeper protected -> public
 
         private final Raider mob;
         private final float hostileRadiusSqr;
