From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Fri, 24 Mar 2023 18:30:34 +0100
Subject: [PATCH] fix paper test plugin


diff --git a/src/test/java/io/papermc/paper/plugin/PaperTestPlugin.java b/src/test/java/io/papermc/paper/plugin/PaperTestPlugin.java
index 1d14f530ef888102e47eeeaf0d1a6076e51871c4..a447dec56bc72e5d7d580063c7d335a2cbe31255 100644
--- a/src/test/java/io/papermc/paper/plugin/PaperTestPlugin.java
+++ b/src/test/java/io/papermc/paper/plugin/PaperTestPlugin.java
@@ -11,10 +11,13 @@ import org.bukkit.plugin.PluginBase;
 import org.bukkit.plugin.PluginDescriptionFile;
 import org.bukkit.plugin.PluginLoader;
 import org.bukkit.plugin.PluginLogger;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.jetbrains.annotations.NotNull;
 
 import java.io.File;
 import java.io.InputStream;
 import java.util.List;
+import java.util.concurrent.Executor;
 
 public class PaperTestPlugin extends PluginBase {
     private final String pluginName;
@@ -91,6 +94,16 @@ public class PaperTestPlugin extends PluginBase {
         throw new UnsupportedOperationException("Not supported.");
     }
 
+    @Override
+    public @NotNull Executor mainThreadExecutor() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
+    @Override
+    public @NonNull Executor asyncExecutor() {
+        throw new UnsupportedOperationException("Not supported.");
+    }
+
     @Override
     public boolean isEnabled() {
         return enabled;
