From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Tue, 21 Jun 2022 20:49:19 +0200
Subject: [PATCH] Make goals public, needs migration to at sometime


diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
index 337a88a7cd6445004d005ef8d56af1b1cdf800d9..2788fd760e16da35fb97b4d08f5a50c4220d1236 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
@@ -697,7 +697,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         return pos.closerThan(this.blockPosition(), (double) distance);
     }
 
-    private class BeePollinateGoal extends Bee.BaseBeeGoal {
+    public class BeePollinateGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         private static final int MIN_POLLINATION_TICKS = 400;
         private static final int MIN_FIND_FLOWER_RETRY_COOLDOWN = 20;
@@ -883,7 +883,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeLookControl extends LookControl {
+    public class BeeLookControl extends LookControl { // Weeper private -> public
 
         BeeLookControl(Mob entity) {
             super(entity);
@@ -919,7 +919,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeEnterHiveGoal extends Bee.BaseBeeGoal {
+    public class BeeEnterHiveGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         BeeEnterHiveGoal() {
             super();
@@ -964,7 +964,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeLocateHiveGoal extends Bee.BaseBeeGoal {
+    public class BeeLocateHiveGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         BeeLocateHiveGoal() {
             super();
@@ -1196,7 +1196,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeGrowCropGoal extends Bee.BaseBeeGoal {
+    public class BeeGrowCropGoal extends Bee.BaseBeeGoal { // Weeper private -> public
 
         static final int GROW_CHANCE = 30;
 
@@ -1264,7 +1264,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeWanderGoal extends Goal {
+    public class BeeWanderGoal extends Goal { // Weeper private -> public
 
         private static final int WANDER_THRESHOLD = 22;
 
@@ -1311,7 +1311,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private class BeeHurtByOtherGoal extends HurtByTargetGoal {
+    public class BeeHurtByOtherGoal extends HurtByTargetGoal { // Weeper private -> public
 
         BeeHurtByOtherGoal(Bee entitybee) {
             super(entitybee);
@@ -1331,7 +1331,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private static class BeeBecomeAngryTargetGoal extends NearestAttackableTargetGoal<Player> {
+    public static class BeeBecomeAngryTargetGoal extends NearestAttackableTargetGoal<Player> { // Weeper private -> public
 
         BeeBecomeAngryTargetGoal(Bee bee) {
             // Objects.requireNonNull(entitybee); // CraftBukkit - decompile error
@@ -1362,7 +1362,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
         }
     }
 
-    private abstract class BaseBeeGoal extends Goal {
+    public abstract class BaseBeeGoal extends Goal { // Weeper private -> public
 
         BaseBeeGoal() {}
 
diff --git a/src/main/java/net/minecraft/world/entity/animal/Cat.java b/src/main/java/net/minecraft/world/entity/animal/Cat.java
index 0114c1cf3b6b0500149a77ebc190cb7fa2832184..0ea8a0d15df8a51b1df340c0276c8708a5465fb0 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Cat.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Cat.java
@@ -509,7 +509,7 @@ public class Cat extends TamableAnimal implements VariantHolder<CatVariant> {
         }
     }
 
-    private static class CatRelaxOnOwnerGoal extends Goal {
+    public static class CatRelaxOnOwnerGoal extends Goal { // Weeper private -> public
 
         private final Cat cat;
         @Nullable
@@ -655,7 +655,7 @@ public class Cat extends TamableAnimal implements VariantHolder<CatVariant> {
         }
     }
 
-    private static class CatAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
+    public static class CatAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> { // Weeper private -> public
 
         private final Cat cat;
 
diff --git a/src/main/java/net/minecraft/world/entity/raid/Raider.java b/src/main/java/net/minecraft/world/entity/raid/Raider.java
index e5ccbaf72f29731f1d1aa939b9297b644a408cd4..4ac37941992d53ce51ea66935ae37eb79b7d075d 100644
--- a/src/main/java/net/minecraft/world/entity/raid/Raider.java
+++ b/src/main/java/net/minecraft/world/entity/raid/Raider.java
@@ -352,7 +352,7 @@ public abstract class Raider extends PatrollingMonster {
         }
     }
 
-    private static class RaiderMoveThroughVillageGoal extends Goal {
+    public static class RaiderMoveThroughVillageGoal extends Goal { // Weeper private -> public
 
         private final Raider raider;
         private final double speedModifier;
@@ -500,7 +500,7 @@ public abstract class Raider extends PatrollingMonster {
         }
     }
 
-    protected class HoldGroundAttackGoal extends Goal {
+    public class HoldGroundAttackGoal extends Goal { // Weeper protected -> public
 
         private final Raider mob;
         private final float hostileRadiusSqr;
