From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Fri, 9 Jun 2023 17:55:57 +0200
Subject: [PATCH] remove duplicatecurly brace


diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index 3fb1ef98ac2c62308dc41b20982c85904306aafd..f7959a65aa41c3ddde7ee79d664cb166fcc32dec 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -106,8 +106,6 @@ public class ServerEntity {
 
                     entityplayer.connection.teleport(entityplayer.getX(), entityplayer.getY(), entityplayer.getZ(), entityplayer.getYRot(), entityplayer.getXRot());
                 }
-                    }
-
             });
             this.lastPassengers = list;
         }
