From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yannick Lamprecht <yannicklamprecht@live.de>
Date: Fri, 9 Jun 2023 17:55:57 +0200
Subject: [PATCH] remove duplicatecurly brace


diff --git a/src/main/java/net/minecraft/server/level/ServerEntity.java b/src/main/java/net/minecraft/server/level/ServerEntity.java
index 094684de6f67c21bdd4d07aa4c6280971ef40791..80bd298f4749052a1e87f89148d9aa0b93adf0e3 100644
--- a/src/main/java/net/minecraft/server/level/ServerEntity.java
+++ b/src/main/java/net/minecraft/server/level/ServerEntity.java
@@ -106,8 +106,6 @@ public class ServerEntity {
 
                     entityplayer.connection.teleport(entityplayer.getX(), entityplayer.getY(), entityplayer.getZ(), entityplayer.getYRot(), entityplayer.getXRot());
                 }
-                    }
-
             });
             this.lastPassengers = list;
         }
